{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"CoLab_TermProj_M3.ipynb","provenance":[],"collapsed_sections":[]},"kernelspec":{"name":"python3","display_name":"Python 3"}},"cells":[{"cell_type":"markdown","metadata":{"id":"_SmTogjeMD47"},"source":["# 생물정보학 및 실습 1 - Term Project용 Mission 3 환경 셋업\n","\n","```\n","생물정보학 및 실습 1\n","서울대학교 협동과정 생물정보학전공 2021년 1학기\n","\n","이 노트북은 Creative Commons BY-SA 조건으로 재사용 가능합니다.\n","장혜식 <hyeshik@snu.ac.kr>, 서울대학교 생명과학부\n","```\n","\n","이 노트북은 Google Colab을 이용해서 Term Project를 할 수 있도록 환경을 준비해 줍니다. 터미널 사용이 원활하지 않은 상태이고, 데이터를 오랫동안 보관하기 어렵기 때문에 되도록 개인용 컴퓨터에서 [Anaconda](https://www.anaconda.com/)나 [JupyterLab](https://jupyterlab.readthedocs.io/en/stable/) 등을 설치하고 사용하시는 것이 편하지만, 개인용 컴퓨터 사양이 부족한 경우에는 Google CoLab을 사용할 수도 있습니다. 개인용 컴퓨터를 쓸 때는 Windows에서는 [WSL2](https://docs.microsoft.com/en-us/windows/wsl/install-win10)를 쓰시면 리눅스 커맨드라인 환경을 쓸 수 있습니다. Google Colab은 창을 닫은 후 90분 후, 또는 창을 처음 열고 12시간 후에 환경을 닫아버리기 때문에 매번 실습할 때마다 새로 셋업이 필요합니다. `Runtime->Run All`을 실행해서 전체를 한 번 실행하시면 준비됩니다. 모든 것이 잘 안 되고 뭔가 꼬인 느낌이 강하게 들 때는 `Runtime->Factory reset runtime`을 누르고 새로 시작하시면 될 수도 있습니다.\n"]},{"cell_type":"markdown","metadata":{"id":"zGYjheLcRJS0"},"source":["## Google Drive 마운트\n","\n","데이터 파일들을 불러오고 저장하기 위해서 Google Drive를 연결합니다. **이번엔 스크립트 파일을 편집해야 하기 때문에 Google Drive 마운트를 하지 않으면 vim등 콘솔에서 쓸 수 있는 에디터를 쓰셔야 합니다.** 화면에 나오는 주소를 눌러서 인증하고 인증 코드를 그 아래에 입력하세요. 그러면 `/contents/drive/MyDrive`에서 자신의 Drive 최상단 폴더가 보입니다."]},{"cell_type":"code","metadata":{"id":"0HgOxANiRHgl","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1622983591721,"user_tz":-540,"elapsed":19688,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"4efa2b5a-c97e-4ed6-c7c6-12e1f5ddf69b"},"source":["from google.colab import drive\n","drive.mount('/content/drive')"],"execution_count":1,"outputs":[{"output_type":"stream","text":["Mounted at /content/drive\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"ra3RB22DWEER"},"source":["## 실습용 Bioconda 환경 설치하기\n","실습용 프로그램들을 쉽게 설치할 수 있도록 [Bioconda](https://bioconda.github.io/) 환경을 만듭니다. 실행하고 진행상황을 관찰하다가 에러로 보이는 것이 있으면 잘 읽어보고 해결합니다.\n","\n","파이썬 모듈들은 Colab쪽의 파이썬 인터프리터를 바꿀 수 없기 때문에 Conda에 설치하더라도 사용하기 어렵습니다. 파이썬 모듈은 `pip` 명령어로 설치하시면 Colab에서 불러와서 사용할 수 있습니다.\n","\n","환경 초기화에 관련된 스크립트는 [GitHub 프로젝트](https://github.com/hyeshik/colab-biolab)에 있습니다."]},{"cell_type":"code","metadata":{"id":"TTXmNi4hcap3","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1622983696303,"user_tz":-540,"elapsed":104592,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"1651b386-57db-4912-8526-ac7ffe161c5e"},"source":["!git clone https://github.com/hyeshik/colab-biolab.git\n","!cd colab-biolab && bash tools/setup.sh\n","exec(open('colab-biolab/tools/activate_conda.py').read())"],"execution_count":2,"outputs":[{"output_type":"stream","text":["Cloning into 'colab-biolab'...\n","remote: Enumerating objects: 69, done.\u001b[K\n","remote: Counting objects: 100% (69/69), done.\u001b[K\n","remote: Compressing objects: 100% (43/43), done.\u001b[K\n","remote: Total 69 (delta 22), reused 54 (delta 13), pack-reused 0\u001b[K\n","Unpacking objects: 100% (69/69), done.\n","./\n","./root/\n","./root/.tmux.conf\n","./root/.bin.priority/\n","./root/.bin.priority/pip3\n","./root/.bin.priority/pip\n","./root/.bin.priority/pip2\n","./root/.profile\n","./root/.bashrc.biolab\n","./root/.condarc\n","./root/.vimrc\n","--2021-06-06 12:46:31--  https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh\n","Resolving repo.anaconda.com (repo.anaconda.com)... 104.16.131.3, 104.16.130.3, 2606:4700::6810:8203, ...\n","Connecting to repo.anaconda.com (repo.anaconda.com)|104.16.131.3|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 94235922 (90M) [application/x-sh]\n","Saving to: ‘miniconda3.sh’\n","\n","miniconda3.sh       100%[===================>]  89.87M   187MB/s    in 0.5s    \n","\n","2021-06-06 12:46:32 (187 MB/s) - ‘miniconda3.sh’ saved [94235922/94235922]\n","\n","PREFIX=/root/conda\n","Unpacking payload ...\n","Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\bdone\n","Solving environment: - \b\b\\ \b\bdone\n","\n","## Package Plan ##\n","\n","  environment location: /root/conda\n","\n","  added / updated specs:\n","    - _libgcc_mutex==0.1=main\n","    - brotlipy==0.7.0=py38h27cfd23_1003\n","    - ca-certificates==2020.10.14=0\n","    - certifi==2020.6.20=pyhd3eb1b0_3\n","    - cffi==1.14.3=py38h261ae71_2\n","    - chardet==3.0.4=py38h06a4308_1003\n","    - conda-package-handling==1.7.2=py38h03888b9_0\n","    - conda==4.9.2=py38h06a4308_0\n","    - cryptography==3.2.1=py38h3c74f83_1\n","    - idna==2.10=py_0\n","    - ld_impl_linux-64==2.33.1=h53a641e_7\n","    - libedit==3.1.20191231=h14c3975_1\n","    - libffi==3.3=he6710b0_2\n","    - libgcc-ng==9.1.0=hdf63c60_0\n","    - libstdcxx-ng==9.1.0=hdf63c60_0\n","    - ncurses==6.2=he6710b0_1\n","    - openssl==1.1.1h=h7b6447c_0\n","    - pip==20.2.4=py38h06a4308_0\n","    - pycosat==0.6.3=py38h7b6447c_1\n","    - pycparser==2.20=py_2\n","    - pyopenssl==19.1.0=pyhd3eb1b0_1\n","    - pysocks==1.7.1=py38h06a4308_0\n","    - python==3.8.5=h7579374_1\n","    - readline==8.0=h7b6447c_0\n","    - requests==2.24.0=py_0\n","    - ruamel_yaml==0.15.87=py38h7b6447c_1\n","    - setuptools==50.3.1=py38h06a4308_1\n","    - six==1.15.0=py38h06a4308_0\n","    - sqlite==3.33.0=h62c20be_0\n","    - tk==8.6.10=hbc83047_0\n","    - tqdm==4.51.0=pyhd3eb1b0_0\n","    - urllib3==1.25.11=py_0\n","    - wheel==0.35.1=pyhd3eb1b0_0\n","    - xz==5.2.5=h7b6447c_0\n","    - yaml==0.2.5=h7b6447c_0\n","    - zlib==1.2.11=h7b6447c_3\n","\n","\n","The following NEW packages will be INSTALLED:\n","\n","  _libgcc_mutex      pkgs/main/linux-64::_libgcc_mutex-0.1-main\n","  brotlipy           pkgs/main/linux-64::brotlipy-0.7.0-py38h27cfd23_1003\n","  ca-certificates    pkgs/main/linux-64::ca-certificates-2020.10.14-0\n","  certifi            pkgs/main/noarch::certifi-2020.6.20-pyhd3eb1b0_3\n","  cffi               pkgs/main/linux-64::cffi-1.14.3-py38h261ae71_2\n","  chardet            pkgs/main/linux-64::chardet-3.0.4-py38h06a4308_1003\n","  conda              pkgs/main/linux-64::conda-4.9.2-py38h06a4308_0\n","  conda-package-han~ pkgs/main/linux-64::conda-package-handling-1.7.2-py38h03888b9_0\n","  cryptography       pkgs/main/linux-64::cryptography-3.2.1-py38h3c74f83_1\n","  idna               pkgs/main/noarch::idna-2.10-py_0\n","  ld_impl_linux-64   pkgs/main/linux-64::ld_impl_linux-64-2.33.1-h53a641e_7\n","  libedit            pkgs/main/linux-64::libedit-3.1.20191231-h14c3975_1\n","  libffi             pkgs/main/linux-64::libffi-3.3-he6710b0_2\n","  libgcc-ng          pkgs/main/linux-64::libgcc-ng-9.1.0-hdf63c60_0\n","  libstdcxx-ng       pkgs/main/linux-64::libstdcxx-ng-9.1.0-hdf63c60_0\n","  ncurses            pkgs/main/linux-64::ncurses-6.2-he6710b0_1\n","  openssl            pkgs/main/linux-64::openssl-1.1.1h-h7b6447c_0\n","  pip                pkgs/main/linux-64::pip-20.2.4-py38h06a4308_0\n","  pycosat            pkgs/main/linux-64::pycosat-0.6.3-py38h7b6447c_1\n","  pycparser          pkgs/main/noarch::pycparser-2.20-py_2\n","  pyopenssl          pkgs/main/noarch::pyopenssl-19.1.0-pyhd3eb1b0_1\n","  pysocks            pkgs/main/linux-64::pysocks-1.7.1-py38h06a4308_0\n","  python             pkgs/main/linux-64::python-3.8.5-h7579374_1\n","  readline           pkgs/main/linux-64::readline-8.0-h7b6447c_0\n","  requests           pkgs/main/noarch::requests-2.24.0-py_0\n","  ruamel_yaml        pkgs/main/linux-64::ruamel_yaml-0.15.87-py38h7b6447c_1\n","  setuptools         pkgs/main/linux-64::setuptools-50.3.1-py38h06a4308_1\n","  six                pkgs/main/linux-64::six-1.15.0-py38h06a4308_0\n","  sqlite             pkgs/main/linux-64::sqlite-3.33.0-h62c20be_0\n","  tk                 pkgs/main/linux-64::tk-8.6.10-hbc83047_0\n","  tqdm               pkgs/main/noarch::tqdm-4.51.0-pyhd3eb1b0_0\n","  urllib3            pkgs/main/noarch::urllib3-1.25.11-py_0\n","  wheel              pkgs/main/noarch::wheel-0.35.1-pyhd3eb1b0_0\n","  xz                 pkgs/main/linux-64::xz-5.2.5-h7b6447c_0\n","  yaml               pkgs/main/linux-64::yaml-0.2.5-h7b6447c_0\n","  zlib               pkgs/main/linux-64::zlib-1.2.11-h7b6447c_3\n","\n","\n","Preparing transaction: / \b\b- \b\b\\ \b\b| \b\bdone\n","Executing transaction: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n","installation finished.\n","WARNING:\n","    You currently have a PYTHONPATH environment variable set. This may cause\n","    unexpected behavior when running the Python interpreter in Miniconda3.\n","    For best results, please verify that your PYTHONPATH only points to\n","    directories of packages that are compatible with the Python interpreter\n","    in Miniconda3: /root/conda\n","Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n","Solving environment: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n","\n","\n","==> WARNING: A newer version of conda exists. <==\n","  current version: 4.9.2\n","  latest version: 4.10.1\n","\n","Please update conda by running\n","\n","    $ conda update -n base -c defaults conda\n","\n","\n","\n","## Package Plan ##\n","\n","  environment location: /root/conda/envs/lab\n","\n","  added / updated specs:\n","    - bedtools\n","    - biopython\n","    - bwa\n","    - cython\n","    - matplotlib\n","    - numpy\n","    - pandas\n","    - pip\n","    - pysam\n","    - samtools\n","    - scipy\n","    - star\n","\n","\n","The following packages will be downloaded:\n","\n","    package                    |            build\n","    ---------------------------|-----------------\n","    _libgcc_mutex-0.1          |      conda_forge           3 KB  conda-forge\n","    _openmp_mutex-4.5          |            1_gnu          22 KB  conda-forge\n","    bedtools-2.30.0            |       h7d7f7ad_1        17.9 MB  bioconda\n","    biopython-1.79             |   py37h5e8e339_0         2.6 MB  conda-forge\n","    bwa-0.7.17                 |       h5bf99c6_8         608 KB  bioconda\n","    bzip2-1.0.8                |       h7f98852_4         484 KB  conda-forge\n","    c-ares-1.17.1              |       h7f98852_1         109 KB  conda-forge\n","    ca-certificates-2021.5.30  |       ha878542_0         136 KB  conda-forge\n","    certifi-2021.5.30          |   py37h89c1867_0         141 KB  conda-forge\n","    cycler-0.10.0              |             py_2           9 KB  conda-forge\n","    cython-0.29.23             |   py37hcd2ae1e_0         2.2 MB  conda-forge\n","    dbus-1.13.6                |       h48d8840_2         572 KB  conda-forge\n","    expat-2.4.1                |       h9c3ff4c_0         182 KB  conda-forge\n","    fontconfig-2.13.1          |    hba837de_1005         357 KB  conda-forge\n","    freetype-2.10.4            |       h0708190_1         890 KB  conda-forge\n","    gettext-0.19.8.1           |    h0b5b191_1005         3.6 MB  conda-forge\n","    glib-2.68.2                |       h9c3ff4c_2         450 KB  conda-forge\n","    glib-tools-2.68.2          |       h9c3ff4c_2          86 KB  conda-forge\n","    gst-plugins-base-1.14.0    |       hbbd80ab_1         4.8 MB\n","    gstreamer-1.14.0           |       h28cd5cc_2         3.2 MB\n","    htslib-1.12                |       h9093b5e_1         2.1 MB  bioconda\n","    icu-58.2                   |    hf484d3e_1000        22.6 MB  conda-forge\n","    jbig-2.1                   |    h7f98852_2003          43 KB  conda-forge\n","    jpeg-9d                    |       h36c2ea0_0         264 KB  conda-forge\n","    kiwisolver-1.3.1           |   py37h2527ec5_1          78 KB  conda-forge\n","    krb5-1.19.1                |       hcc1bbae_0         1.4 MB  conda-forge\n","    lcms2-2.12                 |       hddcbb42_0         443 KB  conda-forge\n","    ld_impl_linux-64-2.35.1    |       hea4e1c9_2         618 KB  conda-forge\n","    lerc-2.2.1                 |       h9c3ff4c_0         213 KB  conda-forge\n","    libblas-3.9.0              |       9_openblas          11 KB  conda-forge\n","    libcblas-3.9.0             |       9_openblas          11 KB  conda-forge\n","    libcurl-7.77.0             |       h2574ce0_0         334 KB  conda-forge\n","    libdeflate-1.7             |       h7f98852_5          67 KB  conda-forge\n","    libedit-3.1.20191231       |       he28a2e2_2         121 KB  conda-forge\n","    libev-4.33                 |       h516909a_1         104 KB  conda-forge\n","    libffi-3.3                 |       h58526e2_2          51 KB  conda-forge\n","    libgcc-ng-9.3.0            |      h2828fa1_19         7.8 MB  conda-forge\n","    libgfortran-ng-9.3.0       |      hff62375_19          22 KB  conda-forge\n","    libgfortran5-9.3.0         |      hff62375_19         2.0 MB  conda-forge\n","    libglib-2.68.2             |       h3e27bee_2         3.1 MB  conda-forge\n","    libgomp-9.3.0              |      h2828fa1_19         376 KB  conda-forge\n","    libiconv-1.16              |       h516909a_0         1.4 MB  conda-forge\n","    liblapack-3.9.0            |       9_openblas          11 KB  conda-forge\n","    libnghttp2-1.43.0          |       h812cca2_0         808 KB  conda-forge\n","    libopenblas-0.3.15         |pthreads_h8fe5266_1         9.2 MB  conda-forge\n","    libpng-1.6.37              |       h21135ba_2         306 KB  conda-forge\n","    libssh2-1.9.0              |       ha56f1ee_6         226 KB  conda-forge\n","    libstdcxx-ng-9.3.0         |      h6de172a_19         4.0 MB  conda-forge\n","    libtiff-4.3.0              |       hf544144_1         668 KB  conda-forge\n","    libuuid-2.32.1             |    h7f98852_1000          28 KB  conda-forge\n","    libwebp-base-1.2.0         |       h7f98852_2         815 KB  conda-forge\n","    libxcb-1.13                |    h7f98852_1003         395 KB  conda-forge\n","    libxml2-2.9.10             |       hb55368b_3         1.2 MB\n","    lz4-c-1.9.3                |       h9c3ff4c_0         179 KB  conda-forge\n","    matplotlib-3.4.2           |   py37h89c1867_0           7 KB  conda-forge\n","    matplotlib-base-3.4.2      |   py37hdd32ed1_0         7.2 MB  conda-forge\n","    ncurses-6.2                |       h58526e2_4         985 KB  conda-forge\n","    numpy-1.20.3               |   py37h038b26d_1         5.7 MB  conda-forge\n","    olefile-0.46               |     pyh9f0ad1d_1          32 KB  conda-forge\n","    openjpeg-2.4.0             |       hb52868f_1         444 KB  conda-forge\n","    openssl-1.1.1k             |       h7f98852_0         2.1 MB  conda-forge\n","    pandas-1.2.4               |   py37h219a48f_0        11.8 MB  conda-forge\n","    pcre-8.44                  |       he1b5a44_0         261 KB  conda-forge\n","    perl-5.32.1                | 0_h7f98852_perl5        14.5 MB  conda-forge\n","    pillow-8.2.0               |   py37h4600e1f_1         684 KB  conda-forge\n","    pip-21.1.2                 |     pyhd8ed1ab_0         1.1 MB  conda-forge\n","    pthread-stubs-0.4          |    h36c2ea0_1001           5 KB  conda-forge\n","    pyparsing-2.4.7            |     pyh9f0ad1d_0          60 KB  conda-forge\n","    pyqt-5.9.2                 |   py37hcca6a23_4         5.7 MB  conda-forge\n","    pysam-0.16.0.1             |   py37h45aed0b_3         2.6 MB  bioconda\n","    python-3.7.10              |hffdb5ce_100_cpython        57.3 MB  conda-forge\n","    python-dateutil-2.8.1      |             py_0         220 KB  conda-forge\n","    python_abi-3.7             |          1_cp37m           4 KB  conda-forge\n","    pytz-2021.1                |     pyhd8ed1ab_0         239 KB  conda-forge\n","    qt-5.9.7                   |       h5867ecd_1        68.5 MB\n","    readline-8.1               |       h46c0cb4_0         295 KB  conda-forge\n","    samtools-1.12              |       h9aed4be_1         386 KB  bioconda\n","    scipy-1.6.3                |   py37h29e03ee_0        20.5 MB  conda-forge\n","    setuptools-49.6.0          |   py37h89c1867_3         947 KB  conda-forge\n","    sip-4.19.8                 |   py37hf484d3e_0         274 KB\n","    six-1.16.0                 |     pyh6c4a22f_0          14 KB  conda-forge\n","    sqlite-3.35.5              |       h74cdb3f_0         1.4 MB  conda-forge\n","    star-2.7.9a                |       h9ee0642_0         3.8 MB  bioconda\n","    tk-8.6.10                  |       h21135ba_1         3.2 MB  conda-forge\n","    tornado-6.1                |   py37h5e8e339_1         646 KB  conda-forge\n","    wheel-0.36.2               |     pyhd3deb0d_0          31 KB  conda-forge\n","    xorg-libxau-1.0.9          |       h7f98852_0          13 KB  conda-forge\n","    xorg-libxdmcp-1.1.3        |       h7f98852_0          19 KB  conda-forge\n","    xz-5.2.5                   |       h516909a_1         343 KB  conda-forge\n","    zlib-1.2.11                |    h516909a_1010         106 KB  conda-forge\n","    zstd-1.5.0                 |       ha95c52a_0         490 KB  conda-forge\n","    ------------------------------------------------------------\n","                                           Total:       310.9 MB\n","\n","The following NEW packages will be INSTALLED:\n","\n","  _libgcc_mutex      conda-forge/linux-64::_libgcc_mutex-0.1-conda_forge\n","  _openmp_mutex      conda-forge/linux-64::_openmp_mutex-4.5-1_gnu\n","  bedtools           bioconda/linux-64::bedtools-2.30.0-h7d7f7ad_1\n","  biopython          conda-forge/linux-64::biopython-1.79-py37h5e8e339_0\n","  bwa                bioconda/linux-64::bwa-0.7.17-h5bf99c6_8\n","  bzip2              conda-forge/linux-64::bzip2-1.0.8-h7f98852_4\n","  c-ares             conda-forge/linux-64::c-ares-1.17.1-h7f98852_1\n","  ca-certificates    conda-forge/linux-64::ca-certificates-2021.5.30-ha878542_0\n","  certifi            conda-forge/linux-64::certifi-2021.5.30-py37h89c1867_0\n","  cycler             conda-forge/noarch::cycler-0.10.0-py_2\n","  cython             conda-forge/linux-64::cython-0.29.23-py37hcd2ae1e_0\n","  dbus               conda-forge/linux-64::dbus-1.13.6-h48d8840_2\n","  expat              conda-forge/linux-64::expat-2.4.1-h9c3ff4c_0\n","  fontconfig         conda-forge/linux-64::fontconfig-2.13.1-hba837de_1005\n","  freetype           conda-forge/linux-64::freetype-2.10.4-h0708190_1\n","  gettext            conda-forge/linux-64::gettext-0.19.8.1-h0b5b191_1005\n","  glib               conda-forge/linux-64::glib-2.68.2-h9c3ff4c_2\n","  glib-tools         conda-forge/linux-64::glib-tools-2.68.2-h9c3ff4c_2\n","  gst-plugins-base   pkgs/main/linux-64::gst-plugins-base-1.14.0-hbbd80ab_1\n","  gstreamer          pkgs/main/linux-64::gstreamer-1.14.0-h28cd5cc_2\n","  htslib             bioconda/linux-64::htslib-1.12-h9093b5e_1\n","  icu                conda-forge/linux-64::icu-58.2-hf484d3e_1000\n","  jbig               conda-forge/linux-64::jbig-2.1-h7f98852_2003\n","  jpeg               conda-forge/linux-64::jpeg-9d-h36c2ea0_0\n","  kiwisolver         conda-forge/linux-64::kiwisolver-1.3.1-py37h2527ec5_1\n","  krb5               conda-forge/linux-64::krb5-1.19.1-hcc1bbae_0\n","  lcms2              conda-forge/linux-64::lcms2-2.12-hddcbb42_0\n","  ld_impl_linux-64   conda-forge/linux-64::ld_impl_linux-64-2.35.1-hea4e1c9_2\n","  lerc               conda-forge/linux-64::lerc-2.2.1-h9c3ff4c_0\n","  libblas            conda-forge/linux-64::libblas-3.9.0-9_openblas\n","  libcblas           conda-forge/linux-64::libcblas-3.9.0-9_openblas\n","  libcurl            conda-forge/linux-64::libcurl-7.77.0-h2574ce0_0\n","  libdeflate         conda-forge/linux-64::libdeflate-1.7-h7f98852_5\n","  libedit            conda-forge/linux-64::libedit-3.1.20191231-he28a2e2_2\n","  libev              conda-forge/linux-64::libev-4.33-h516909a_1\n","  libffi             conda-forge/linux-64::libffi-3.3-h58526e2_2\n","  libgcc-ng          conda-forge/linux-64::libgcc-ng-9.3.0-h2828fa1_19\n","  libgfortran-ng     conda-forge/linux-64::libgfortran-ng-9.3.0-hff62375_19\n","  libgfortran5       conda-forge/linux-64::libgfortran5-9.3.0-hff62375_19\n","  libglib            conda-forge/linux-64::libglib-2.68.2-h3e27bee_2\n","  libgomp            conda-forge/linux-64::libgomp-9.3.0-h2828fa1_19\n","  libiconv           conda-forge/linux-64::libiconv-1.16-h516909a_0\n","  liblapack          conda-forge/linux-64::liblapack-3.9.0-9_openblas\n","  libnghttp2         conda-forge/linux-64::libnghttp2-1.43.0-h812cca2_0\n","  libopenblas        conda-forge/linux-64::libopenblas-0.3.15-pthreads_h8fe5266_1\n","  libpng             conda-forge/linux-64::libpng-1.6.37-h21135ba_2\n","  libssh2            conda-forge/linux-64::libssh2-1.9.0-ha56f1ee_6\n","  libstdcxx-ng       conda-forge/linux-64::libstdcxx-ng-9.3.0-h6de172a_19\n","  libtiff            conda-forge/linux-64::libtiff-4.3.0-hf544144_1\n","  libuuid            conda-forge/linux-64::libuuid-2.32.1-h7f98852_1000\n","  libwebp-base       conda-forge/linux-64::libwebp-base-1.2.0-h7f98852_2\n","  libxcb             conda-forge/linux-64::libxcb-1.13-h7f98852_1003\n","  libxml2            pkgs/main/linux-64::libxml2-2.9.10-hb55368b_3\n","  lz4-c              conda-forge/linux-64::lz4-c-1.9.3-h9c3ff4c_0\n","  matplotlib         conda-forge/linux-64::matplotlib-3.4.2-py37h89c1867_0\n","  matplotlib-base    conda-forge/linux-64::matplotlib-base-3.4.2-py37hdd32ed1_0\n","  ncurses            conda-forge/linux-64::ncurses-6.2-h58526e2_4\n","  numpy              conda-forge/linux-64::numpy-1.20.3-py37h038b26d_1\n","  olefile            conda-forge/noarch::olefile-0.46-pyh9f0ad1d_1\n","  openjpeg           conda-forge/linux-64::openjpeg-2.4.0-hb52868f_1\n","  openssl            conda-forge/linux-64::openssl-1.1.1k-h7f98852_0\n","  pandas             conda-forge/linux-64::pandas-1.2.4-py37h219a48f_0\n","  pcre               conda-forge/linux-64::pcre-8.44-he1b5a44_0\n","  perl               conda-forge/linux-64::perl-5.32.1-0_h7f98852_perl5\n","  pillow             conda-forge/linux-64::pillow-8.2.0-py37h4600e1f_1\n","  pip                conda-forge/noarch::pip-21.1.2-pyhd8ed1ab_0\n","  pthread-stubs      conda-forge/linux-64::pthread-stubs-0.4-h36c2ea0_1001\n","  pyparsing          conda-forge/noarch::pyparsing-2.4.7-pyh9f0ad1d_0\n","  pyqt               conda-forge/linux-64::pyqt-5.9.2-py37hcca6a23_4\n","  pysam              bioconda/linux-64::pysam-0.16.0.1-py37h45aed0b_3\n","  python             conda-forge/linux-64::python-3.7.10-hffdb5ce_100_cpython\n","  python-dateutil    conda-forge/noarch::python-dateutil-2.8.1-py_0\n","  python_abi         conda-forge/linux-64::python_abi-3.7-1_cp37m\n","  pytz               conda-forge/noarch::pytz-2021.1-pyhd8ed1ab_0\n","  qt                 pkgs/main/linux-64::qt-5.9.7-h5867ecd_1\n","  readline           conda-forge/linux-64::readline-8.1-h46c0cb4_0\n","  samtools           bioconda/linux-64::samtools-1.12-h9aed4be_1\n","  scipy              conda-forge/linux-64::scipy-1.6.3-py37h29e03ee_0\n","  setuptools         conda-forge/linux-64::setuptools-49.6.0-py37h89c1867_3\n","  sip                pkgs/main/linux-64::sip-4.19.8-py37hf484d3e_0\n","  six                conda-forge/noarch::six-1.16.0-pyh6c4a22f_0\n","  sqlite             conda-forge/linux-64::sqlite-3.35.5-h74cdb3f_0\n","  star               bioconda/linux-64::star-2.7.9a-h9ee0642_0\n","  tk                 conda-forge/linux-64::tk-8.6.10-h21135ba_1\n","  tornado            conda-forge/linux-64::tornado-6.1-py37h5e8e339_1\n","  wheel              conda-forge/noarch::wheel-0.36.2-pyhd3deb0d_0\n","  xorg-libxau        conda-forge/linux-64::xorg-libxau-1.0.9-h7f98852_0\n","  xorg-libxdmcp      conda-forge/linux-64::xorg-libxdmcp-1.1.3-h7f98852_0\n","  xz                 conda-forge/linux-64::xz-5.2.5-h516909a_1\n","  zlib               conda-forge/linux-64::zlib-1.2.11-h516909a_1010\n","  zstd               conda-forge/linux-64::zstd-1.5.0-ha95c52a_0\n","\n","\n","\n","Downloading and Extracting Packages\n","ca-certificates-2021 | 136 KB    | : 100% 1.0/1 [00:00<00:00, 15.53it/s]\n","glib-tools-2.68.2    | 86 KB     | : 100% 1.0/1 [00:00<00:00, 21.34it/s]\n","xorg-libxau-1.0.9    | 13 KB     | : 100% 1.0/1 [00:00<00:00, 33.64it/s]\n","pillow-8.2.0         | 684 KB    | : 100% 1.0/1 [00:00<00:00,  4.83it/s]\n","pandas-1.2.4         | 11.8 MB   | : 100% 1.0/1 [00:03<00:00,  3.06s/it]\n","setuptools-49.6.0    | 947 KB    | : 100% 1.0/1 [00:00<00:00,  3.35it/s]\n","pthread-stubs-0.4    | 5 KB      | : 100% 1.0/1 [00:00<00:00, 33.37it/s]\n","libblas-3.9.0        | 11 KB     | : 100% 1.0/1 [00:00<00:00, 26.39it/s]\n","bedtools-2.30.0      | 17.9 MB   | : 100% 1.0/1 [00:03<00:00,  3.30s/it]               \n","libedit-3.1.20191231 | 121 KB    | : 100% 1.0/1 [00:00<00:00, 21.75it/s]\n","libtiff-4.3.0        | 668 KB    | : 100% 1.0/1 [00:00<00:00,  6.64it/s]\n","tornado-6.1          | 646 KB    | : 100% 1.0/1 [00:00<00:00,  4.77it/s]\n","pysam-0.16.0.1       | 2.6 MB    | : 100% 1.0/1 [00:00<00:00,  2.02it/s]\n","libgfortran5-9.3.0   | 2.0 MB    | : 100% 1.0/1 [00:00<00:00,  2.27it/s]\n","libdeflate-1.7       | 67 KB     | : 100% 1.0/1 [00:00<00:00, 27.86it/s]\n","pyparsing-2.4.7      | 60 KB     | : 100% 1.0/1 [00:00<00:00, 31.75it/s]\n","libstdcxx-ng-9.3.0   | 4.0 MB    | : 100% 1.0/1 [00:00<00:00,  1.42it/s]\n","libwebp-base-1.2.0   | 815 KB    | : 100% 1.0/1 [00:00<00:00,  5.73it/s]\n","libgfortran-ng-9.3.0 | 22 KB     | : 100% 1.0/1 [00:00<00:00, 33.65it/s]\n","olefile-0.46         | 32 KB     | : 100% 1.0/1 [00:00<00:00, 32.85it/s]\n","glib-2.68.2          | 450 KB    | : 100% 1.0/1 [00:00<00:00,  5.22it/s]\n","star-2.7.9a          | 3.8 MB    | : 100% 1.0/1 [00:00<00:00,  1.43it/s]\n","certifi-2021.5.30    | 141 KB    | : 100% 1.0/1 [00:00<00:00, 20.22it/s]\n","pip-21.1.2           | 1.1 MB    | : 100% 1.0/1 [00:00<00:00,  2.87it/s]\n","libuuid-2.32.1       | 28 KB     | : 100% 1.0/1 [00:00<00:00, 31.35it/s]\n","jbig-2.1             | 43 KB     | : 100% 1.0/1 [00:00<00:00, 29.12it/s]\n","lcms2-2.12           | 443 KB    | : 100% 1.0/1 [00:00<00:00,  8.12it/s]\n","cython-0.29.23       | 2.2 MB    | : 100% 1.0/1 [00:00<00:00,  1.72it/s]\n","python_abi-3.7       | 4 KB      | : 100% 1.0/1 [00:00<00:00, 33.36it/s]\n","qt-5.9.7             | 68.5 MB   | : 100% 1.0/1 [00:03<00:00,  3.07s/it]               \n","matplotlib-3.4.2     | 7 KB      | : 100% 1.0/1 [00:00<00:00, 23.47it/s]\n","scipy-1.6.3          | 20.5 MB   | : 100% 1.0/1 [00:04<00:00,  4.42s/it]\n","expat-2.4.1          | 182 KB    | : 100% 1.0/1 [00:00<00:00, 12.84it/s]\n","freetype-2.10.4      | 890 KB    | : 100% 1.0/1 [00:00<00:00,  4.90it/s]\n","zlib-1.2.11          | 106 KB    | : 100% 1.0/1 [00:00<00:00, 22.82it/s]\n","ncurses-6.2          | 985 KB    | : 100% 1.0/1 [00:00<00:00,  2.10it/s]\n","libiconv-1.16        | 1.4 MB    | : 100% 1.0/1 [00:00<00:00,  4.43it/s]\n","cycler-0.10.0        | 9 KB      | : 100% 1.0/1 [00:00<00:00, 39.46it/s]\n","libgcc-ng-9.3.0      | 7.8 MB    | : 100% 1.0/1 [00:01<00:00,  1.40s/it]\n","samtools-1.12        | 386 KB    | : 100% 1.0/1 [00:00<00:00,  9.31it/s]\n","tk-8.6.10            | 3.2 MB    | : 100% 1.0/1 [00:00<00:00,  1.33it/s]\n","six-1.16.0           | 14 KB     | : 100% 1.0/1 [00:00<00:00, 28.38it/s]\n","ld_impl_linux-64-2.3 | 618 KB    | : 100% 1.0/1 [00:00<00:00,  5.92it/s]\n","fontconfig-2.13.1    | 357 KB    | : 100% 1.0/1 [00:00<00:00,  9.24it/s]\n","kiwisolver-1.3.1     | 78 KB     | : 100% 1.0/1 [00:00<00:00, 20.76it/s]\n","readline-8.1         | 295 KB    | : 100% 1.0/1 [00:00<00:00, 10.71it/s]\n","bwa-0.7.17           | 608 KB    | : 100% 1.0/1 [00:00<00:00,  6.59it/s]\n","libcurl-7.77.0       | 334 KB    | : 100% 1.0/1 [00:00<00:00, 12.00it/s]\n","libssh2-1.9.0        | 226 KB    | : 100% 1.0/1 [00:00<00:00, 11.73it/s]\n","biopython-1.79       | 2.6 MB    | : 100% 1.0/1 [00:00<00:00,  1.13it/s]\n","jpeg-9d              | 264 KB    | : 100% 1.0/1 [00:00<00:00, 10.49it/s]\n","openjpeg-2.4.0       | 444 KB    | : 100% 1.0/1 [00:00<00:00,  7.99it/s]\n","sip-4.19.8           | 274 KB    | : 100% 1.0/1 [00:00<00:00, 12.52it/s]\n","liblapack-3.9.0      | 11 KB     | : 100% 1.0/1 [00:00<00:00, 30.79it/s]\n","icu-58.2             | 22.6 MB   | : 100% 1.0/1 [00:04<00:00,  4.24s/it]\n","gstreamer-1.14.0     | 3.2 MB    | : 100% 1.0/1 [00:00<00:00,  5.89it/s]\n","_libgcc_mutex-0.1    | 3 KB      | : 100% 1.0/1 [00:00<00:00, 33.12it/s]\n","gettext-0.19.8.1     | 3.6 MB    | : 100% 1.0/1 [00:01<00:00,  1.04s/it]\n","libgomp-9.3.0        | 376 KB    | : 100% 1.0/1 [00:00<00:00, 11.17it/s]\n","libcblas-3.9.0       | 11 KB     | : 100% 1.0/1 [00:00<00:00, 36.07it/s]\n","libopenblas-0.3.15   | 9.2 MB    | : 100% 1.0/1 [00:01<00:00,  1.99s/it]               \n","zstd-1.5.0           | 490 KB    | : 100% 1.0/1 [00:00<00:00,  8.46it/s]\n","libev-4.33           | 104 KB    | : 100% 1.0/1 [00:00<00:00, 15.84it/s]\n","numpy-1.20.3         | 5.7 MB    | : 100% 1.0/1 [00:01<00:00,  1.52s/it]\n","sqlite-3.35.5        | 1.4 MB    | : 100% 1.0/1 [00:00<00:00,  3.82it/s]\n","libpng-1.6.37        | 306 KB    | : 100% 1.0/1 [00:00<00:00, 10.15it/s]\n","c-ares-1.17.1        | 109 KB    | : 100% 1.0/1 [00:00<00:00, 15.67it/s]\n","_openmp_mutex-4.5    | 22 KB     | : 100% 1.0/1 [00:00<00:00, 24.52it/s]\n","gst-plugins-base-1.1 | 4.8 MB    | : 100% 1.0/1 [00:00<00:00,  4.47it/s]\n","python-3.7.10        | 57.3 MB   | : 100% 1.0/1 [00:08<00:00,  8.95s/it]               \n","pytz-2021.1          | 239 KB    | : 100% 1.0/1 [00:00<00:00,  7.55it/s]\n","openssl-1.1.1k       | 2.1 MB    | : 100% 1.0/1 [00:00<00:00,  2.47it/s]\n","xorg-libxdmcp-1.1.3  | 19 KB     | : 100% 1.0/1 [00:00<00:00, 37.37it/s]\n","wheel-0.36.2         | 31 KB     | : 100% 1.0/1 [00:00<00:00, 24.35it/s]\n","python-dateutil-2.8. | 220 KB    | : 100% 1.0/1 [00:00<00:00, 19.34it/s]\n","perl-5.32.1          | 14.5 MB   | : 100% 1.0/1 [00:03<00:00,  3.96s/it]\n","libnghttp2-1.43.0    | 808 KB    | : 100% 1.0/1 [00:00<00:00,  5.93it/s]\n","pcre-8.44            | 261 KB    | : 100% 1.0/1 [00:00<00:00, 13.22it/s]\n","libxcb-1.13          | 395 KB    | : 100% 1.0/1 [00:00<00:00,  6.30it/s]\n","lerc-2.2.1           | 213 KB    | : 100% 1.0/1 [00:00<00:00, 14.10it/s]\n","libxml2-2.9.10       | 1.2 MB    | : 100% 1.0/1 [00:00<00:00,  7.03it/s]\n","lz4-c-1.9.3          | 179 KB    | : 100% 1.0/1 [00:00<00:00, 13.81it/s]\n","dbus-1.13.6          | 572 KB    | : 100% 1.0/1 [00:00<00:00,  6.81it/s]\n","pyqt-5.9.2           | 5.7 MB    | : 100% 1.0/1 [00:01<00:00,  1.46s/it]               \n","xz-5.2.5             | 343 KB    | : 100% 1.0/1 [00:00<00:00,  8.07it/s]\n","libglib-2.68.2       | 3.1 MB    | : 100% 1.0/1 [00:00<00:00,  1.34it/s]\n","matplotlib-base-3.4. | 7.2 MB    | : 100% 1.0/1 [00:01<00:00,  1.62s/it]               \n","bzip2-1.0.8          | 484 KB    | : 100% 1.0/1 [00:00<00:00,  6.55it/s]\n","libffi-3.3           | 51 KB     | : 100% 1.0/1 [00:00<00:00, 30.45it/s]\n","htslib-1.12          | 2.1 MB    | : 100% 1.0/1 [00:00<00:00,  2.42it/s]\n","krb5-1.19.1          | 1.4 MB    | : 100% 1.0/1 [00:00<00:00,  3.30it/s]\n","Preparing transaction: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n","Verifying transaction: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n","Executing transaction: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n","#\n","# To activate this environment, use\n","#\n","#     $ conda activate lab\n","#\n","# To deactivate an active environment, use\n","#\n","#     $ conda deactivate\n","\n","Activated conda environment `lab'!\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"eDfJ0fCzsTTb"},"source":["설정이 끝났으면, 이제 콘솔을 열어서 터미널로 접속할 수 있는 통로를 열어줍니다."]},{"cell_type":"markdown","metadata":{"id":"J2LqGiP_N26s"},"source":["## Google Drive에 시퀀싱 데이터 다운로드\n","이제 실습 수업에서 필요한 시퀀싱 데이터를 다운로드 받아서 설치합시다. 드라이브 공간만 충분하면 데이터는 계속 저장돼 있기 때문에 한 번만 하시면 됩니다. Colab에서는 작업용 데이터를 Google Drive 폴더 아래의 `binfo-datapack1`에 저장합니다. 실습용 데이터는 `/content/drive/MyDrive/binfo1-datapack1`에 풀리게 됩니다. 첫 번째에는 wget이 접속오류를 내는 경우가 많으니, 접속에러가 발생하면 다시 한 번 실행해 보세요. 1G씩 끊어지지만 자동으로 이어받아집니다."]},{"cell_type":"code","metadata":{"id":"kIN484e7NSxR"},"source":["!wget -O - --no-check-certificate https://hyeshik.qbio.io/binfo/binfo1-datapack1.tar | tar -C /content/drive/MyDrive -xf -"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"lnlM-S70C-tY"},"source":["받은 데이터의 MD5 체크섬을 확인합니다. 한참 걸립니다. 각 파일들의 원래 체크섬은 다음과 같습니다. 한 번 비교해 보세요.\n","\n","```\n","140aaf30bcb9276cc716f8699f04ddd6  CLIP-35L33G.bam\n","f1b3336ed7e2f97d562dcc71641251bd  CLIP-35L33G.bam.bai\n","328883a73d507eafbf5b60bd6b906201  RNA-control.bam\n","02073818e2f398a73c3b76e5169de1ca  RNA-control.bam.bai\n","b09550d09d6c2a4ce27f0226f426fdb1  RNA-siLin28a.bam\n","fef112c727244060ea62d3f2564a07f6  RNA-siLin28a.bam.bai\n","28bbd0c47d725669340c784f1b772c01  RNA-siLuc.bam\n","43590fdc4d81905c0432e0d1cb8cfd5b  RNA-siLuc.bam.bai\n","5c08a9297307bc83259e658c4474f0cc  RPF-siLin28a.bam\n","a1bb3e29be412dfd7fd8d16b1b1acc4c  RPF-siLin28a.bam.bai\n","f2eebf50943024d0116c9cd3e744c707  RPF-siLuc.bam\n","dc24f69e8f571fc8be30f28ce5b84fcd  RPF-siLuc.bam.bai\n","```"]},{"cell_type":"code","metadata":{"id":"g8lC_YJRBCPf"},"source":["!md5sum drive/MyDrive/binfo1-datapack1/*"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"kiQ5y4dTD27M"},"source":["그리고, GENCODE Annotation 데이터도 미리 받아서 저장해 둡니다. 에러가 나면 다시 한 번 실행해 봅니다."]},{"cell_type":"code","metadata":{"id":"Efdd3x8hDMh4"},"source":["!wget --no-check-certificate -O /content/drive/MyDrive/binfo1-datapack1/gencode.gtf.gz http://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M27/gencode.vM27.annotation.gtf.gz"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"O3rM9ds6EMhB"},"source":["다루기 쉽도록 Annotation 압축을 풉니다."]},{"cell_type":"code","metadata":{"id":"KNyuSM6PEQvX"},"source":["!gunzip /content/drive/MyDrive/binfo1-datapack1/gencode.gtf.gz"],"execution_count":null,"outputs":[]},{"cell_type":"code","metadata":{"id":"PVXzIs5FE3Q4"},"source":["!ls -al drive/MyDrive/binfo*"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"ecEA1PPQUlIK"},"source":["실습에 필요한 프로그램들 몇 가지를 추가로 설치합니다."]},{"cell_type":"code","metadata":{"id":"neteyWsWEKBy","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1622983704353,"user_tz":-540,"elapsed":8054,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"f5174a2e-e4d4-4b7f-d52d-1651c3be35f8"},"source":["!conda install -y bedtools bioawk"],"execution_count":3,"outputs":[{"output_type":"stream","text":["Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n","Solving environment: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n","\n","\n","==> WARNING: A newer version of conda exists. <==\n","  current version: 4.9.2\n","  latest version: 4.10.1\n","\n","Please update conda by running\n","\n","    $ conda update -n base -c defaults conda\n","\n","\n","\n","## Package Plan ##\n","\n","  environment location: /root/conda/envs/lab\n","\n","  added / updated specs:\n","    - bedtools\n","    - bioawk\n","\n","\n","The following packages will be downloaded:\n","\n","    package                    |            build\n","    ---------------------------|-----------------\n","    bioawk-1.0                 |       h5bf99c6_6         188 KB  bioconda\n","    ------------------------------------------------------------\n","                                           Total:         188 KB\n","\n","The following NEW packages will be INSTALLED:\n","\n","  bioawk             bioconda/linux-64::bioawk-1.0-h5bf99c6_6\n","\n","\n","\n","Downloading and Extracting Packages\n","bioawk-1.0           | 188 KB    | : 100% 1.0/1 [00:00<00:00,  8.12it/s]\n","Preparing transaction: | \b\bdone\n","Verifying transaction: - \b\bdone\n","Executing transaction: | \b\bdone\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"Oe_54MZrEi4y"},"source":["## Mission 3\n","\n","이번 미션에서는 Figure S2A 아랫쪽에 표시되는 clustered error 분석 (CIMS; Crosslinking-induced mutation sites)을 합니다. 자외선에 의해서 단백질이 붙어있다가 단백질 분해효소로 단백질을 뜯어내고 남은 자리에서는 reversetranscriptase가 오류를 자주 냅니다. 오류율을 분석해서 단백질이 앉아 있었던 자리를 분석해 봅시다.\n","\n","데이터는 지난 번과 동일하기 때문에 이미 다운로드 받으셨다면 다시 다운로드 받을 필요는 없습니다."]},{"cell_type":"code","metadata":{"id":"IANz6DmLUkA6","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1622983714811,"user_tz":-540,"elapsed":268,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"92b9bc0b-5975-45d2-dc08-cc4e1670b503"},"source":["%cd /content/drive/MyDrive/Colab_Notebooks/binfo1-datapack1/"],"execution_count":4,"outputs":[{"output_type":"stream","text":["/content/drive/MyDrive/Colab_Notebooks/binfo1-datapack1\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"CMMDdHgyJAB1"},"source":["Genome 전체로 계산하는 방법은 자원도 꽤 쓰고 시간이 걸리기 때문에 여기서는 유전자 단위로 간략하게 하도록 합시다. 예시로 Mirlet7g를 해 보고, 추가로 Mirlet7f-1과 Mirlet7d도 봐서 비교해 봅시다.\n","\n","논문은 예전 레퍼런스 유전체인 mm9을 쓰고 있기 때문에, 현재 사용되는 레퍼런스 mm39와는 좌표계에 차이가 있습니다. 새로 annotation에서 위치를 찾습니다."]},{"cell_type":"code","metadata":{"id":"sumX23OcI_fl","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1622983734718,"user_tz":-540,"elapsed":11732,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"63861137-8407-40ee-bdea-c313922c92aa"},"source":["!grep -i mirlet7g gencode.gtf"],"execution_count":5,"outputs":[{"output_type":"stream","text":["chr9\tENSEMBL\tgene\t106056039\t106056126\t.\t+\t.\tgene_id \"ENSMUSG00000065440.3\"; gene_type \"miRNA\"; gene_name \"Mirlet7g\"; level 3; mgi_id \"MGI:2676800\";\n","chr9\tENSEMBL\ttranscript\t106056039\t106056126\t.\t+\t.\tgene_id \"ENSMUSG00000065440.3\"; transcript_id \"ENSMUST00000083506.3\"; gene_type \"miRNA\"; gene_name \"Mirlet7g\"; transcript_type \"miRNA\"; transcript_name \"Mirlet7g-201\"; level 3; transcript_support_level \"NA\"; mgi_id \"MGI:2676800\"; tag \"basic\";\n","chr9\tENSEMBL\texon\t106056039\t106056126\t.\t+\t.\tgene_id \"ENSMUSG00000065440.3\"; transcript_id \"ENSMUST00000083506.3\"; gene_type \"miRNA\"; gene_name \"Mirlet7g\"; transcript_type \"miRNA\"; transcript_name \"Mirlet7g-201\"; exon_number 1; exon_id \"ENSMUSE00000522665.2\"; level 3; transcript_support_level \"NA\"; mgi_id \"MGI:2676800\"; tag \"basic\";\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"iB_C8P1fJkNE"},"source":["Splicing 되지 않는 transcript이기 때문에 결과가 단조롭습니다. 해당 좌표 범위로 bam을 솎아냅니다."]},{"cell_type":"code","metadata":{"id":"opqsj9vNIL-z","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1622983737021,"user_tz":-540,"elapsed":2307,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"8a1e8aec-43e5-434c-8d72-c25fb618be2a"},"source":["!samtools view -b -o CLIP-let7g.bam CLIP-35L33G.bam chr9:106056039-106056126\n","!samtools view CLIP-let7g.bam | wc -l"],"execution_count":6,"outputs":[{"output_type":"stream","text":["[W::hts_idx_load3] The index file is older than the data file: CLIP-35L33G.bam.bai\n","163\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"baEtdt6aJ12N"},"source":["각 자리에 mapping된 read들의 분포를 요약하기 위해 samtools mpileup을 이용합니다. Shannon entropy를 구할 때는 reference를 따로 줄 필요는 없습니다. samtools를 쓰지 않고 직접 read를 분석하면 더욱 더 재미있습니다."]},{"cell_type":"code","metadata":{"id":"5_RLXcMpIL8J","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1622983737409,"user_tz":-540,"elapsed":400,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"7784b9a3-6743-40e9-aecd-e787ce199db4"},"source":["!samtools mpileup CLIP-let7g.bam > CLIP-let7g.pileup\n","!wc -l CLIP-let7g.pileup"],"execution_count":7,"outputs":[{"output_type":"stream","text":["[mpileup] 1 samples in 1 input files\n","68548 CLIP-let7g.pileup\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"Xuzv5lSEKVoG"},"source":["Mirlet7g는 100bp도 안 되는 짧은 gene인데 `mpileup` 결과는 엄청 길게 나옵니다. 왜 그럴까요? (잘 한 번 생각해 봅시다)"]},{"cell_type":"markdown","metadata":{"id":"hpC-bsZEKozF"},"source":["이제 Mirlet7g의 핵심부분을 살펴봐서 어떻게 하면 좋을지 견적을 내 봅니다."]},{"cell_type":"code","metadata":{"id":"1WulVeZqIL5W","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1622983737681,"user_tz":-540,"elapsed":274,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"3917ebf6-0928-473e-837a-91fa1076c0de"},"source":["!head CLIP-let7g.pileup"],"execution_count":8,"outputs":[{"output_type":"stream","text":["chr9\t106007092\tN\t9\t^Ga^Ia^Ha^Ia^Ga^Ga^Ha^Ia^Ia\t<IFIGGHII\n","chr9\t106007093\tN\t9\tggggggggg\tAFGIEGDII\n","chr9\t106007094\tN\t9\tccccccccc\tEHHIGAFEH\n","chr9\t106007095\tN\t9\taaaaaaaaa\t?HDIFA>II\n","chr9\t106007096\tN\t9\taaaaaaaaa\tEHHIG@CHH\n","chr9\t106007097\tN\t9\tttttttttt\t=HGEGDFII\n","chr9\t106007098\tN\t9\taaaaaaaaa\t?HHIGDEII\n","chr9\t106007099\tN\t9\tg-2nng-2nng-2nng-2nng-2nng-2nng-2nng-2nng-2nn\t?IEHGEDII\n","chr9\t106007100\tN\t9\t*********\t8IBIBC@EI\n","chr9\t106007101\tN\t9\t*********\t8IBIBC@EI\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"GUdB31wAIL0K","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1622983737958,"user_tz":-540,"elapsed":278,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"4ddd5571-f793-49f0-d5e2-91771687b70e"},"source":["!awk '$2 >= 106056039 && $2 <= 106056126 { print $0; }' CLIP-let7g.pileup > CLIP-let7g-gene.pileup\n","!tail CLIP-let7g-gene.pileup"],"execution_count":9,"outputs":[{"output_type":"stream","text":["chr9\t106056117\tN\t138\t<<<<<<<<<CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC\tBHEIG?DIIHGHII:;HGIIDGGGIIBGEB?>H<GGIIIIDIIGFHIHH9IGG=GGHIDH?DGIBIIHHGGIEIGI8GIDHDG.GGHEDIIDIDGDHIDGIFGHG;DCDDHEE@I?CGG:IHGIBGIHIIDG@DHIGG\n","chr9\t106056118\tN\t139\t<<<<<<<<<CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC\tBHEIG?DIIHGHII3AHIIIDGGGIIB3EHFBH>GIIIIIIIIIHIIHHAIIGBDGHIDH?GIIGIIHHDIIHIGIGGI=BGG9GIHHHIIGI=IAIIGGIEHGIEBG@DHDE;IBDGG4IGGIGGIIIIGGGDGIH1G\n","chr9\t106056119\tN\t138\t<<<<<<<<<TTTTTTTTTTTTTTTTTTTTTTTTTT$TTTT$TTTTTTTTTTTTTTT$TTTT$TTTTTTT$TTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTTT\tBHEIG?DIIGEFII:/GIIEDDHDIIF>BH3<HBEGIHIGFIIIBIIBE:HHGD>EGGDH7BIIGGIHHDHIHDEIGEIDHGG;GIHGHHIGD:IGHD:GGEHH?3G87HD</I=DGG=IHHIHBIFII@<HDHHH1G\n","chr9\t106056120\tN\t133\t<<<<<<<<<TTTTTTTTTTTTTTTTTTTTTTTTTTTT$TTTT$TTT$TTTTTTTTTT$TTTTTTTT$TTTTTTTTTTTT$T$TTTTTT$TTTT$TT$TTTTTTTTTTTTTTTTTTT$TTTTTTTTTTTTTTTTTTTTTTTTTTT\tBHEIG?DIIHFBII74BIIIGDIIIIA8DHECHBIIIIIIIIHIIHB?IDGGGFDH?GIIIIHH@GHHGEDDGIEHBEDIHHGIIGI:HBII8DIIHHIBDD;<HDD3HB?@G?HHIIHIIIIIB@HDHDH8G\n","chr9\t106056121\tN\t118\t<<<<<<<<<GGGGGGGGGGGGGGGGGGGG$G$G$G$G$G$G$G$GGGGGG$GG$GGGG$GG$GGGG$G$GGG$GGG$GGG$GGGG$GGGGG$GGGGGGGGG$GGGGGG$GG$GGG$GGG$GGGG$G$GG$GGGGGG$GGGG$G$G$G$G$G$\tBHEIG?DIIB<EDI/7BHHFB<FGHI3?GG>HAIHIFEF@H<<2>D3EE;HBAIIIBH4IHG@DI>G482GGFHID:I9II@BH@>HE8@;<HGI:GGBIDEIBIGIIIE?G@EIH7G\n","chr9\t106056122\tN\t88\t<<<<<<<<<C$C$C$C$C$C$C$C$C$C$C$C$C$CCCCCC$C$C$C$C$C$C$CCC$C$C$C$CC$C$CC$CC$C$C$C$C$C$C$C$C$C$C$C$C$C$CC$C$C$C$C$C$CC$CC$C$CCC$C$C$C$C$CCCCC$CCC$C\tBHEIG?DIIEEGIIC;GIHEGBIIIIB1=FII?FEIGGGHDBIG=HHIIEBHD>EIHIG=I:HGBIIEF?E;DB3HB<GIHIHHEADF\n","chr9\t106056123\tN\t31\t<<<<<<<<<CCCCCCCCCCCCCCCCCCCCCC\tBHEIG?DIIIIII>GIGGIGGD>BIHHHIEH\n","chr9\t106056124\tN\t31\t<<<<<<<<<AAAAAAAAAAAAAAAAAAAAAA\tBHEIG?DIIIIHIGGIGGGIG:9DDBIEGFH\n","chr9\t106056125\tN\t31\t<<<<<<<<<GGGGGGGGGGGGGGGGGGGGGG\tBHEIG?DIIIIIIGGE@GFIGD;GIGIIFHD\n","chr9\t106056126\tN\t30\t<<<<<<<<<GGGGGGGGGGGGGGGGGGGGG\tBHEIG?DIIIIGHGHIGHI>G;GGGIGIHG\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"CNEWC4t_MA__"},"source":["이제 pandas에 불러와서 분석해 봅니다. R을 쓰시는 분들은 R로 보내서 작업하셔도 됩니다."]},{"cell_type":"code","metadata":{"id":"PWq1ItOxILxa","colab":{"base_uri":"https://localhost:8080/","height":223},"executionInfo":{"status":"ok","timestamp":1622983738604,"user_tz":-540,"elapsed":10,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"9688378f-241d-4489-f827-1cd69f4e0a56"},"source":["import pandas as pd\n","\n","pileup = pd.read_csv('CLIP-let7g-gene.pileup', sep='\\t', names=['chrom', 'pos', '_ref', 'count', 'basereads', 'quals'])\n","pileup.tail()"],"execution_count":10,"outputs":[{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>chrom</th>\n","      <th>pos</th>\n","      <th>_ref</th>\n","      <th>count</th>\n","      <th>basereads</th>\n","      <th>quals</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>83</th>\n","      <td>chr9</td>\n","      <td>106056122</td>\n","      <td>N</td>\n","      <td>88</td>\n","      <td>&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;C$C$C$C$C$C$C$C$C$C$C$C$C$CCCCCC$C$C$...</td>\n","      <td>BHEIG?DIIEEGIIC;GIHEGBIIIIB1=FII?FEIGGGHDBIG=H...</td>\n","    </tr>\n","    <tr>\n","      <th>84</th>\n","      <td>chr9</td>\n","      <td>106056123</td>\n","      <td>N</td>\n","      <td>31</td>\n","      <td>&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;CCCCCCCCCCCCCCCCCCCCCC</td>\n","      <td>BHEIG?DIIIIII&gt;GIGGIGGD&gt;BIHHHIEH</td>\n","    </tr>\n","    <tr>\n","      <th>85</th>\n","      <td>chr9</td>\n","      <td>106056124</td>\n","      <td>N</td>\n","      <td>31</td>\n","      <td>&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;AAAAAAAAAAAAAAAAAAAAAA</td>\n","      <td>BHEIG?DIIIIHIGGIGGGIG:9DDBIEGFH</td>\n","    </tr>\n","    <tr>\n","      <th>86</th>\n","      <td>chr9</td>\n","      <td>106056125</td>\n","      <td>N</td>\n","      <td>31</td>\n","      <td>&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;GGGGGGGGGGGGGGGGGGGGGG</td>\n","      <td>BHEIG?DIIIIIIGGE@GFIGD;GIGIIFHD</td>\n","    </tr>\n","    <tr>\n","      <th>87</th>\n","      <td>chr9</td>\n","      <td>106056126</td>\n","      <td>N</td>\n","      <td>30</td>\n","      <td>&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;&lt;GGGGGGGGGGGGGGGGGGGGG</td>\n","      <td>BHEIG?DIIIIGHGHIGHI&gt;G;GGGIGIHG</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["   chrom  ...                                              quals\n","83  chr9  ...  BHEIG?DIIEEGIIC;GIHEGBIIIIB1=FII?FEIGGGHDBIG=H...\n","84  chr9  ...                    BHEIG?DIIIIII>GIGGIGGD>BIHHHIEH\n","85  chr9  ...                    BHEIG?DIIIIHIGGIGGGIG:9DDBIEGFH\n","86  chr9  ...                    BHEIG?DIIIIIIGGE@GFIGD;GIGIIFHD\n","87  chr9  ...                     BHEIG?DIIIIGHGHIGHI>G;GGGIGIHG\n","\n","[5 rows x 6 columns]"]},"metadata":{"tags":[]},"execution_count":10}]},{"cell_type":"markdown","metadata":{"id":"DOM-v48vMTQ-"},"source":["여기서 pileup에서 계산에 사용할 것은 match와 substitution만입니다. 나머지에 해당하는 tag들은 제거합니다.\n","\n","R을 쓰시는 분들은 R로 보내서 작업하셔도 됩니다."]},{"cell_type":"code","metadata":{"id":"qm25CsvBILug","executionInfo":{"status":"ok","timestamp":1622983742325,"user_tz":-540,"elapsed":400,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}}},"source":["import re\n","toremove = re.compile('[<>$*#^]')\n","pileup['matches'] = pileup['basereads'].apply(lambda x: toremove.sub('', x))"],"execution_count":11,"outputs":[]},{"cell_type":"code","metadata":{"id":"uXYF___MILrK","colab":{"base_uri":"https://localhost:8080/","height":422},"executionInfo":{"status":"ok","timestamp":1622983746029,"user_tz":-540,"elapsed":284,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"d42d7f43-ba84-413e-8ee0-29bb56c4e0ea"},"source":["pileup[['chrom', 'pos', 'matches']]"],"execution_count":12,"outputs":[{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>chrom</th>\n","      <th>pos</th>\n","      <th>matches</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>chr9</td>\n","      <td>106056039</td>\n","      <td></td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>chr9</td>\n","      <td>106056040</td>\n","      <td></td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>chr9</td>\n","      <td>106056041</td>\n","      <td></td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>chr9</td>\n","      <td>106056042</td>\n","      <td></td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>chr9</td>\n","      <td>106056043</td>\n","      <td></td>\n","    </tr>\n","    <tr>\n","      <th>...</th>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","    </tr>\n","    <tr>\n","      <th>83</th>\n","      <td>chr9</td>\n","      <td>106056122</td>\n","      <td>CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC...</td>\n","    </tr>\n","    <tr>\n","      <th>84</th>\n","      <td>chr9</td>\n","      <td>106056123</td>\n","      <td>CCCCCCCCCCCCCCCCCCCCCC</td>\n","    </tr>\n","    <tr>\n","      <th>85</th>\n","      <td>chr9</td>\n","      <td>106056124</td>\n","      <td>AAAAAAAAAAAAAAAAAAAAAA</td>\n","    </tr>\n","    <tr>\n","      <th>86</th>\n","      <td>chr9</td>\n","      <td>106056125</td>\n","      <td>GGGGGGGGGGGGGGGGGGGGGG</td>\n","    </tr>\n","    <tr>\n","      <th>87</th>\n","      <td>chr9</td>\n","      <td>106056126</td>\n","      <td>GGGGGGGGGGGGGGGGGGGGG</td>\n","    </tr>\n","  </tbody>\n","</table>\n","<p>88 rows × 3 columns</p>\n","</div>"],"text/plain":["   chrom        pos                                            matches\n","0   chr9  106056039                                                   \n","1   chr9  106056040                                                   \n","2   chr9  106056041                                                   \n","3   chr9  106056042                                                   \n","4   chr9  106056043                                                   \n","..   ...        ...                                                ...\n","83  chr9  106056122  CCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCCC...\n","84  chr9  106056123                             CCCCCCCCCCCCCCCCCCCCCC\n","85  chr9  106056124                             AAAAAAAAAAAAAAAAAAAAAA\n","86  chr9  106056125                             GGGGGGGGGGGGGGGGGGGGGG\n","87  chr9  106056126                              GGGGGGGGGGGGGGGGGGGGG\n","\n","[88 rows x 3 columns]"]},"metadata":{"tags":[]},"execution_count":12}]},{"cell_type":"code","metadata":{"id":"k55NUm9OILhw","colab":{"base_uri":"https://localhost:8080/","height":53},"executionInfo":{"status":"ok","timestamp":1622987254899,"user_tz":-540,"elapsed":257,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"fab26afa-92d8-4ff2-811b-a6dd0b9de003"},"source":["pileup[pileup['pos'] ==106056093].iloc[0]['matches']"],"execution_count":92,"outputs":[{"output_type":"execute_result","data":{"application/vnd.google.colaboratory.intrinsic+json":{"type":"string"},"text/plain":["'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAHAIAHAIAGAHAIAGAHAGABAHADAGAIAHAHAHAIAIAHAIAGAIAGAIAIAGAHAIAIAHAHAIAGAHAGADADAHAGAGAIAGADAGAGA'"]},"metadata":{"tags":[]},"execution_count":92}]},{"cell_type":"code","metadata":{"id":"5dhUe_DIYPkx","executionInfo":{"status":"ok","timestamp":1622988698923,"user_tz":-540,"elapsed":256,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}}},"source":["import numpy as np\n","\n","base_key=['A','T','G','C']\n","shanon_ent=[]"],"execution_count":122,"outputs":[]},{"cell_type":"code","metadata":{"id":"mOVZhOKslsrC","executionInfo":{"status":"ok","timestamp":1622988699167,"user_tz":-540,"elapsed":3,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}}},"source":["for i in pileup['pos']:\n","  matched_base=pileup[pileup['pos'] == i].iloc[0]['matches']\n","  total_num_of_base=len(matched_base)\n","  freq_dict={'A':0,'T':0,'G':0,'C':0}\n","  shanon_dict={'A':0,'T':0,'G':0,'C':0}\n","  if total_num_of_base!=0:\n","    for j in base_key:\n","      freq_dict[j]=matched_base.count(j)/total_num_of_base\n","      if freq_dict[j]==0:\n","        shanon_dict[j]=0\n","      else:\n","        shanon_dict[j]=freq_dict[j]*np.log(freq_dict[j])\n","    shanon_ent.append(-sum(shanon_dict.values()))\n","  else:\n","    shanon_ent.append(0)"],"execution_count":123,"outputs":[]},{"cell_type":"code","metadata":{"id":"9MC_J-qwowQp","executionInfo":{"status":"ok","timestamp":1622988701658,"user_tz":-540,"elapsed":304,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}}},"source":["pos_p_1=pileup['pos']+1"],"execution_count":124,"outputs":[]},{"cell_type":"code","metadata":{"id":"0MgiA6RlpnwK","executionInfo":{"status":"ok","timestamp":1622988701915,"user_tz":-540,"elapsed":8,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}}},"source":["rst_bedform=pd.concat([pileup[['chrom', 'pos']],pos_p_1,pd.DataFrame(shanon_ent)],axis=1)"],"execution_count":125,"outputs":[]},{"cell_type":"code","metadata":{"id":"1pYZNStyqPaV","executionInfo":{"status":"ok","timestamp":1622992817499,"user_tz":-540,"elapsed":408,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}}},"source":["rst_bedform.to_csv(\"bed_form.csv\",sep=\" \",header=None,index=False)"],"execution_count":134,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"NrgJiS2iRhIn"},"source":["자 이제 데이터 준비가 대충 끝났습니다. 다음 순서로 진행해 보세요. (1-4번 단계는 R을 쓰시는 분들은 R로 보내서 작업하셔도 됩니다.)\n","\n","1. 각 position별로 base수를 셉니다.\n","1. 각 position별로 Shannon entropy를 계산합니다.\n","1. 계산한 Shannon entropy를 [bedgraph format](https://genome.ucsc.edu/goldenPath/help/bedgraph.html)으로 출력합니다. 말은 복잡해도 실제로는 간단합니다. 4칸짜리를 만드시면 됩니다.\n","1. 결과 파일을 본인의 Google Drive에서 가져옵니다.\n","1. [UCSC Genome Browser](http://genome.ucsc.edu/cgi-bin/hgTracks?db=mm39&lastVirtModeType=default&lastVirtModeExtraState=&virtModeType=default&virtMode=0&nonVirtPosition=&position=chr9%3A106056039-106056126)에 접속해서 Genome은 mm39로 선택합니다.\n","1. 그래프 아랫쪽의 add custom tracks 버튼을 누릅니다.\n","1. Paste URLs or data 옆의 Choose File 버튼을 누르고 우리가 만든 bedgraph 파일을 업로드 합니다.\n","1. 그리고 이리 저리 감상해 보다가 View 메뉴의 PDF/PS 를 눌러서 인증샷을 한 번 찍습니다.\n","1. Mirlet7d와 Mirlet7f-1도 한 번 해 봅니다."]},{"cell_type":"markdown","metadata":{"id":"k43UyjXt_PHE"},"source":["# Mirlet7d"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"B3mpdM8zsi7j","executionInfo":{"status":"ok","timestamp":1622993921354,"user_tz":-540,"elapsed":10310,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"a7bfd65c-e397-466e-89a5-057094a6ade2"},"source":["!grep -i Mirlet7d gencode.gtf"],"execution_count":135,"outputs":[{"output_type":"stream","text":["chr13\tENSEMBL\tgene\t48689488\t48689590\t.\t-\t.\tgene_id \"ENSMUSG00000065453.3\"; gene_type \"miRNA\"; gene_name \"Mirlet7d\"; level 3; mgi_id \"MGI:2676796\";\n","chr13\tENSEMBL\ttranscript\t48689488\t48689590\t.\t-\t.\tgene_id \"ENSMUSG00000065453.3\"; transcript_id \"ENSMUST00000083519.3\"; gene_type \"miRNA\"; gene_name \"Mirlet7d\"; transcript_type \"miRNA\"; transcript_name \"Mirlet7d-201\"; level 3; transcript_support_level \"NA\"; mgi_id \"MGI:2676796\"; tag \"basic\";\n","chr13\tENSEMBL\texon\t48689488\t48689590\t.\t-\t.\tgene_id \"ENSMUSG00000065453.3\"; transcript_id \"ENSMUST00000083519.3\"; gene_type \"miRNA\"; gene_name \"Mirlet7d\"; transcript_type \"miRNA\"; transcript_name \"Mirlet7d-201\"; exon_number 1; exon_id \"ENSMUSE00000522678.2\"; level 3; transcript_support_level \"NA\"; mgi_id \"MGI:2676796\"; tag \"basic\";\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"J4Df5bpt-bth","executionInfo":{"status":"ok","timestamp":1622994043366,"user_tz":-540,"elapsed":1065,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"22edc871-9d5d-4a00-bb58-01fcf662ac6e"},"source":["!samtools view -b -o CLIP-let7d.bam CLIP-35L33G.bam chr13:48689488-48689590\n","!samtools view CLIP-let7d.bam | wc -l"],"execution_count":136,"outputs":[{"output_type":"stream","text":["[W::hts_idx_load3] The index file is older than the data file: CLIP-35L33G.bam.bai\n","187\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"FdCPEmeT-7wo","executionInfo":{"status":"ok","timestamp":1622994078087,"user_tz":-540,"elapsed":282,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"88bcda66-2f79-4c8c-e8db-67893ec7dd6e"},"source":["!samtools mpileup CLIP-let7d.bam > CLIP-let7d.pileup\n","!wc -l CLIP-let7d.pileup"],"execution_count":137,"outputs":[{"output_type":"stream","text":["[mpileup] 1 samples in 1 input files\n","87 CLIP-let7d.pileup\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"8-gDcDe4_Ebg","executionInfo":{"status":"ok","timestamp":1622994102122,"user_tz":-540,"elapsed":287,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"73134055-05ff-45ee-e725-5e984914682c"},"source":["!head CLIP-let7d.pileup"],"execution_count":138,"outputs":[{"output_type":"stream","text":["chr13\t48689488\tN\t24\t^Ga^Ia^Ha^Ia^Ha^Ia^Ia^Ia^Ga^Ia^Ia^Ia^Ia^Ia^Ha^Ia^Ga^Ga^Ga^Ia^Ea^Da^Ha^Ga\tBIHGGIHIEIIIGIHIGEGH5BDD\n","chr13\t48689489\tN\t53\taaaaaaaaaaaaaaaaaaaaaaaaa^Ia^Ha^Ia^Ga^Ga^Ha^Ia^Ia^Ga^Ga^Ia^Ia^Ha^Ia^Ia^Ha^Ha^Da^Ia^Ha^Ia^Ia^Ia^Ia^Ia^Ha^Ga^Ga\tGIHIGIIIEIIIIIHIG>EB4@5HDIHIBEGIID?IIGII@H>IGHIIIGHDC\n","chr13\t48689490\tN\t69\tttttttttttttttttttttttttttttttttttttttttttttttttttttt^Gt^Gt^Ht^Gt^It^It^Ht^It^Gt^Gt^Ht^It^It^It^Ht^It\tGIHIGHIFFIIIIIFHGEDG;2>FDIHIDGBIIDBIIDIIAH>IBGIIIGHBG=BG=IIGIDGHFIIHI\n","chr13\t48689491\tN\t105\taaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa^Ia^Ia^Ha^Ga^Ia^Ga^Ia^Ga^Ha^Ha^Ha^Ga^Ia^Ha^Ha^Ia^@a^Ia^Ia^Ia^Ga^Ha^Ha^Ga^Ha^Ga^Ia^Ha^Ga^Ia^Ga^Ha^Ga^Ia^Ia^Ga\tEIGHEHHIEIIIIIHIGCBH:?>HGGHI>GBIIGBIIGIHBB;IHIIIIGHDC3=G:IGGIDEHIFIGHII=GIDHEGHHEIGGI7III9=HGBAIGBIGFG>I2\n","chr13\t48689492\tN\t105\taaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa\tGIGIEIEI7IIIGIGIG;?F:D@G7GHIAGAIIG3IIGIHDB2IHIIIIDDGC1<GCIIHIGCHIIIEIIIGCIBIEGHH?HHFI9IIIA=GGB?IEGIFEGGIA\n","chr13\t48689493\tN\t105\tggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggg\tBIEIGHGI7IHHGIGHG=?F58>E5EHIAGAHIDAHIGII:EDHHIHIIGG>>5EDAIIFFGDHIIHGIII8EIDIBBHHBIGHI7HII?=GGBEIG>HFEGGIA\n","chr13\t48689494\tN\t105\tggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggg\tFIFEGIFE3HFIGIGHG=?D1?@D5EHIAEADIBFIIGHIBD<IHDGIIGBD>03GEEIHF@DHIIIDIIHEAIDIC>HHBHFDI9FIIE=GGBAIGEFEEGIIA\n","chr13\t48689495\tN\t104\tcccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc\tGID>GHEG.HIHGH>IG2;B;?>E0BCI8G?FGA?HH@HIB?DHBIIIHGBG90BAEIEF8CEDIHGIIHGAI?I3BGH;HHHE7IIIA=GG:?IFEBF2GBI;\n","chr13\t48689496\tN\t102\tcccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc\tFI8EEGEI.GEB<ID>G38B19>BFFI<D<FGAEFIGDFB@@CH>EHH8B7>BDCEEEBBCHBGEEFHIG?I;IG@HH2EEGI9IIIA>;<@IE<E:CDIG<\n","chr13\t48689497\tN\t104\tcccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc\tDIHFBIFG.HIIFIGIB3;B993E:IGHAGDIDC3IIHDIBDHBIIIII>;58?A>IIEHBAHBHIGHIEFBI0BG<HHEEGDG.III?7H?>3IE<H8EDGG?\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"gfGpGbyN_KTG","executionInfo":{"status":"ok","timestamp":1622994190249,"user_tz":-540,"elapsed":297,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"696422d9-716a-4252-f4b1-a6fbed9cf387"},"source":["!awk '$2 >= 48689488 && $2 <= 48689590 { print $0; }' CLIP-let7d.pileup > CLIP-let7d-gene.pileup\n","!tail CLIP-let7d-gene.pileup"],"execution_count":141,"outputs":[{"output_type":"stream","text":["chr13\t48689565\tN\t6\tcccccc\t>IGIIH\n","chr13\t48689566\tN\t6\ttttttt\t:IGFIH\n","chr13\t48689567\tN\t6\taaaaaa\t8IFIFI\n","chr13\t48689568\tN\t6\tcccccc\t?IGIII\n","chr13\t48689569\tN\t6\ttttttt\t;IGIII\n","chr13\t48689570\tN\t5\taaaaa\tEDECD\n","chr13\t48689571\tN\t6\tcccccc\t:GGIGG\n","chr13\t48689572\tN\t6\tcccccc\t:GGIGG\n","chr13\t48689573\tN\t6\ttttttt\t5GGIGG\n","chr13\t48689574\tN\t6\tc$c$c$c$c$c$\t;GGIGG\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":205},"id":"PePaX77b_T3-","executionInfo":{"status":"ok","timestamp":1622994192609,"user_tz":-540,"elapsed":264,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"40570ce9-0bfc-4206-a6ca-c6ab37cb9acc"},"source":["pileup = pd.read_csv('CLIP-let7d-gene.pileup', sep='\\t', names=['chrom', 'pos', '_ref', 'count', 'basereads', 'quals'])\n","pileup.tail()"],"execution_count":142,"outputs":[{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>chrom</th>\n","      <th>pos</th>\n","      <th>_ref</th>\n","      <th>count</th>\n","      <th>basereads</th>\n","      <th>quals</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>82</th>\n","      <td>chr13</td>\n","      <td>48689570</td>\n","      <td>N</td>\n","      <td>5</td>\n","      <td>aaaaa</td>\n","      <td>EDECD</td>\n","    </tr>\n","    <tr>\n","      <th>83</th>\n","      <td>chr13</td>\n","      <td>48689571</td>\n","      <td>N</td>\n","      <td>6</td>\n","      <td>cccccc</td>\n","      <td>:GGIGG</td>\n","    </tr>\n","    <tr>\n","      <th>84</th>\n","      <td>chr13</td>\n","      <td>48689572</td>\n","      <td>N</td>\n","      <td>6</td>\n","      <td>cccccc</td>\n","      <td>:GGIGG</td>\n","    </tr>\n","    <tr>\n","      <th>85</th>\n","      <td>chr13</td>\n","      <td>48689573</td>\n","      <td>N</td>\n","      <td>6</td>\n","      <td>tttttt</td>\n","      <td>5GGIGG</td>\n","    </tr>\n","    <tr>\n","      <th>86</th>\n","      <td>chr13</td>\n","      <td>48689574</td>\n","      <td>N</td>\n","      <td>6</td>\n","      <td>c$c$c$c$c$c$</td>\n","      <td>;GGIGG</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["    chrom       pos _ref  count     basereads   quals\n","82  chr13  48689570    N      5         aaaaa   EDECD\n","83  chr13  48689571    N      6        cccccc  :GGIGG\n","84  chr13  48689572    N      6        cccccc  :GGIGG\n","85  chr13  48689573    N      6        tttttt  5GGIGG\n","86  chr13  48689574    N      6  c$c$c$c$c$c$  ;GGIGG"]},"metadata":{"tags":[]},"execution_count":142}]},{"cell_type":"code","metadata":{"id":"75y-x3rUAWev","executionInfo":{"status":"ok","timestamp":1622994419096,"user_tz":-540,"elapsed":257,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}}},"source":["toremove = re.compile('[<>$*#^]')\n","pileup['matches'] = pileup['basereads'].apply(lambda x: toremove.sub('', x))"],"execution_count":144,"outputs":[]},{"cell_type":"code","metadata":{"id":"q_yL3_2DADS1","executionInfo":{"status":"ok","timestamp":1622994420484,"user_tz":-540,"elapsed":380,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}}},"source":["shanon_ent=[]\n","for i in pileup['pos']:\n","  matched_base=pileup[pileup['pos'] == i].iloc[0]['matches']\n","  total_num_of_base=len(matched_base)\n","  freq_dict={'A':0,'T':0,'G':0,'C':0}\n","  shanon_dict={'A':0,'T':0,'G':0,'C':0}\n","  if total_num_of_base!=0:\n","    for j in base_key:\n","      freq_dict[j]=matched_base.count(j)/total_num_of_base\n","      if freq_dict[j]==0:\n","        shanon_dict[j]=0\n","      else:\n","        shanon_dict[j]=freq_dict[j]*np.log(freq_dict[j])\n","    shanon_ent.append(-sum(shanon_dict.values()))\n","  else:\n","    shanon_ent.append(0)\n","\n","pos_p_1=pileup['pos']+1\n","rst_bedform=pd.concat([pileup[['chrom', 'pos']],pos_p_1,pd.DataFrame(shanon_ent)],axis=1)\n","rst_bedform.to_csv(\"bed_form7d.csv\",sep=\" \",header=None,index=False)"],"execution_count":145,"outputs":[]},{"cell_type":"code","metadata":{"id":"lMeQzvOy_YZE","executionInfo":{"status":"ok","timestamp":1622994452769,"user_tz":-540,"elapsed":240,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}}},"source":[""],"execution_count":145,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"tI_7YpyDAgRg"},"source":["# Mirlet7f-1"]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"xR5EO4xhAgRh","executionInfo":{"status":"ok","timestamp":1622994611266,"user_tz":-540,"elapsed":2706,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"e7ae5d53-ffc1-498b-ecf9-05169d5348e8"},"source":["!grep -i Mirlet7f-1 gencode.gtf"],"execution_count":148,"outputs":[{"output_type":"stream","text":["chr13\tENSEMBL\tgene\t48691305\t48691393\t.\t-\t.\tgene_id \"ENSMUSG00000105621.3\"; gene_type \"miRNA\"; gene_name \"Mirlet7f-1\"; level 3; mgi_id \"MGI:2676798\";\n","chr13\tENSEMBL\ttranscript\t48691305\t48691393\t.\t-\t.\tgene_id \"ENSMUSG00000105621.3\"; transcript_id \"ENSMUST00000198652.3\"; gene_type \"miRNA\"; gene_name \"Mirlet7f-1\"; transcript_type \"miRNA\"; transcript_name \"Mirlet7f-1-201\"; level 3; transcript_support_level \"NA\"; mgi_id \"MGI:2676798\"; tag \"basic\";\n","chr13\tENSEMBL\texon\t48691305\t48691393\t.\t-\t.\tgene_id \"ENSMUSG00000105621.3\"; transcript_id \"ENSMUST00000198652.3\"; gene_type \"miRNA\"; gene_name \"Mirlet7f-1\"; transcript_type \"miRNA\"; transcript_name \"Mirlet7f-1-201\"; exon_number 1; exon_id \"ENSMUSE00001348852.2\"; level 3; transcript_support_level \"NA\"; mgi_id \"MGI:2676798\"; tag \"basic\";\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"K_BTAopTAgRi","executionInfo":{"status":"ok","timestamp":1622994619276,"user_tz":-540,"elapsed":393,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"edcaf7d4-feda-4ec1-8a26-25582065b704"},"source":["!samtools view -b -o CLIP-letf_1.bam CLIP-35L33G.bam chr13:48691305-48691393\n","!samtools view CLIP-letf_1.bam | wc -l"],"execution_count":149,"outputs":[{"output_type":"stream","text":["[W::hts_idx_load3] The index file is older than the data file: CLIP-35L33G.bam.bai\n","158\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"RSaokiGxAgRi","executionInfo":{"status":"ok","timestamp":1622994623602,"user_tz":-540,"elapsed":598,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"55e234ce-3649-4973-cbf4-675c25497089"},"source":["!samtools mpileup CLIP-letf_1.bam > CLIP-letf_1.pileup\n","!wc -l CLIP-letf_1.pileup"],"execution_count":150,"outputs":[{"output_type":"stream","text":["[mpileup] 1 samples in 1 input files\n","97 CLIP-letf_1.pileup\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"NiyvvdGSAgRi","executionInfo":{"status":"ok","timestamp":1622994627383,"user_tz":-540,"elapsed":466,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"93b8a828-5af1-4e8f-96ff-843b29b59137"},"source":["!head CLIP-letf_1.pileup"],"execution_count":151,"outputs":[{"output_type":"stream","text":["chr13\t48691290\tN\t1\t^Gg\tA\n","chr13\t48691291\tN\t26\tc^Hc^Ic^Dc^Hc^Gc^Ic^Gc^Ic^Gc^Ic^Ic^Hc^Hc^Hc^Ec^Ic^Ic^Ic^Hc^Hc^Hc^Ic^Ic^Ic^Hc\tDDI4G>I@H;GGGFH7IHIHHHFIIG\n","chr13\t48691292\tN\t25\taaaaaaaaaaaaaaaaaaaaaaaaa\tDGIG>IBI?IIGHH7HIEHFFGHIG\n","chr13\t48691293\tN\t38\tgggggggggggggggggggggggggg^Ig^Hg^Ig^Ig^Ig^Ig^Hg^Ig^Ig^Dg^Ig^Ig\tDGI4>AIGIBIIGGE/IIIDGHIIIHEGIHIIBIIBII\n","chr13\t48691294\tN\t49\tcccccccccccccccccccccccccccccccccccccc^Ec^Hc^Ic^Hc^Hc^Gc^Ic^Ic^Gc^Ic^Hc\t?>I1E5IDG<I>EHH7IIIFHHEIHGCEDIHIEHG?II3BIDD7IHGGH\n","chr13\t48691295\tN\t58\taaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa^Ha^Ia^Ha^Ha^Ha^Ia^Ia^Ia^Ha\tBDI?HBIGHGIHGHG7IIIGBGIIIGBEFIIIBIH=II=EIGA6IHGIHDIHGGIII0\n","chr13\t48691296\tN\t81\taaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa^Ia^Ia^Da^Ia^Ha^Ia^Ha^Ga^Ha^Ha^Ga^Ia^Ha^Ha^Ia^Ha^Ha^Ia^Ha^Ia^Ga^Ia^Ga\t<GI7H?IGIEIIGHG;FIIGGHIIIGIBGDIHGII@II9FHDA9IIGGHGIEGGIIIBEIBIHIG;DG?IBHICGIHI6I;\n","chr13\t48691297\tN\t80\tgggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggggg\t8EI.HBIGIGIGGHG1IIIGGGIIIEIBGGIIDII=II=FIGD5IIGDHDIHGGIGH@DIDGDIB1GGDIHGD=EHHGH5\n","chr13\t48691298\tN\t80\ttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttttt\tGI4H7IGIGIIGHG7IHIGDHIGIGIBGGHIDIH@II?EIGG=IIGIHDGHG8IFH7DH@HHIG1GH9HDHI<GIDB.H/\n","chr13\t48691299\tN\t80\tcccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc\t<<I7H=I@I;IHGGG7HIIGGHGGHGFEBIEIDII=II:HIC><IIGIHGHGEGIII;HGBIHGD;GF<IGHICEIDII4\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"yBwTk8btAgRj","executionInfo":{"status":"ok","timestamp":1622994636652,"user_tz":-540,"elapsed":333,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"5a0a5958-7014-4e99-a7fc-e715acf3ab5e"},"source":["!awk '$2 >= 48691305 && $2 <= 48691393 { print $0; }' CLIP-letf_1.pileup > CLIP-letf_1-gene.pileup\n","!tail CLIP-letf_1-gene.pileup"],"execution_count":152,"outputs":[{"output_type":"stream","text":["chr13\t48691377\tN\t24\ttttttttttttttttttttttttt\tBIIGIGEDCGFIFGI@GHIDDIGI\n","chr13\t48691378\tN\t24\taaaaaaaaaaaaaaaaaaaaaaaa\t?HIFIGGGAFEFABI>GII;EIGI\n","chr13\t48691379\tN\t24\tcccccccccccccccccccccccc\tEIIGIGGDEGGIEGI;GIIDDIGI\n","chr13\t48691380\tN\t24\ttttttttttttttttttttttttt\tEIIGIBGGEGFIFGI@GIIBDIGI\n","chr13\t48691381\tN\t24\taaaaaaaaaaaaaaaaaaaaaaaa\tBEHFIGEG@FCEAGH4GII;AIDH\n","chr13\t48691382\tN\t24\tcccccccccccccccccccccccc\t?IEGIHGG?GEGEBI=GIIDEIGG\n","chr13\t48691383\tN\t24\tcccccccccccccccccccccccc\tEIGGIGGGAGFGFGI=GIIDGIGG\n","chr13\t48691384\tN\t24\ttttttttttttttttttttttttt\t<IGGIHGG>GEGBGI8GII@GIGG\n","chr13\t48691385\tN\t24\tcccccccccccccccccccccccc\tBIGGIBDD?GEGBEI=GIIBGIGG\n","chr13\t48691386\tN\t24\ta$a$a$a$a$a$a$a$a$a$a$a$a$a$a$a$a$a$a$a$a$a$a$a$\tEIGGIFDECGFGFGI2BIIDGIGG\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":205},"id":"kk1gn_wFAgRj","executionInfo":{"status":"ok","timestamp":1622994639498,"user_tz":-540,"elapsed":233,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"06afacda-e00e-4311-ebf3-7ae677f01f74"},"source":["pileup = pd.read_csv('CLIP-letf_1-gene.pileup', sep='\\t', names=['chrom', 'pos', '_ref', 'count', 'basereads', 'quals'])\n","pileup.tail()"],"execution_count":153,"outputs":[{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>chrom</th>\n","      <th>pos</th>\n","      <th>_ref</th>\n","      <th>count</th>\n","      <th>basereads</th>\n","      <th>quals</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>77</th>\n","      <td>chr13</td>\n","      <td>48691382</td>\n","      <td>N</td>\n","      <td>24</td>\n","      <td>cccccccccccccccccccccccc</td>\n","      <td>?IEGIHGG?GEGEBI=GIIDEIGG</td>\n","    </tr>\n","    <tr>\n","      <th>78</th>\n","      <td>chr13</td>\n","      <td>48691383</td>\n","      <td>N</td>\n","      <td>24</td>\n","      <td>cccccccccccccccccccccccc</td>\n","      <td>EIGGIGGGAGFGFGI=GIIDGIGG</td>\n","    </tr>\n","    <tr>\n","      <th>79</th>\n","      <td>chr13</td>\n","      <td>48691384</td>\n","      <td>N</td>\n","      <td>24</td>\n","      <td>tttttttttttttttttttttttt</td>\n","      <td>&lt;IGGIHGG&gt;GEGBGI8GII@GIGG</td>\n","    </tr>\n","    <tr>\n","      <th>80</th>\n","      <td>chr13</td>\n","      <td>48691385</td>\n","      <td>N</td>\n","      <td>24</td>\n","      <td>cccccccccccccccccccccccc</td>\n","      <td>BIGGIBDD?GEGBEI=GIIBGIGG</td>\n","    </tr>\n","    <tr>\n","      <th>81</th>\n","      <td>chr13</td>\n","      <td>48691386</td>\n","      <td>N</td>\n","      <td>24</td>\n","      <td>a$a$a$a$a$a$a$a$a$a$a$a$a$a$a$a$a$a$a$a$a$a$a$a$</td>\n","      <td>EIGGIFDECGFGFGI2BIIDGIGG</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["    chrom  ...                     quals\n","77  chr13  ...  ?IEGIHGG?GEGEBI=GIIDEIGG\n","78  chr13  ...  EIGGIGGGAGFGFGI=GIIDGIGG\n","79  chr13  ...  <IGGIHGG>GEGBGI8GII@GIGG\n","80  chr13  ...  BIGGIBDD?GEGBEI=GIIBGIGG\n","81  chr13  ...  EIGGIFDECGFGFGI2BIIDGIGG\n","\n","[5 rows x 6 columns]"]},"metadata":{"tags":[]},"execution_count":153}]},{"cell_type":"code","metadata":{"id":"ktPD2OZnAgRj","executionInfo":{"status":"ok","timestamp":1622994641992,"user_tz":-540,"elapsed":357,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}}},"source":["toremove = re.compile('[<>$*#^]')\n","pileup['matches'] = pileup['basereads'].apply(lambda x: toremove.sub('', x))"],"execution_count":154,"outputs":[]},{"cell_type":"code","metadata":{"id":"rUS2Hk6cAgRk","executionInfo":{"status":"ok","timestamp":1622994644070,"user_tz":-540,"elapsed":312,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}}},"source":["shanon_ent=[]\n","for i in pileup['pos']:\n","  matched_base=pileup[pileup['pos'] == i].iloc[0]['matches']\n","  total_num_of_base=len(matched_base)\n","  freq_dict={'A':0,'T':0,'G':0,'C':0}\n","  shanon_dict={'A':0,'T':0,'G':0,'C':0}\n","  if total_num_of_base!=0:\n","    for j in base_key:\n","      freq_dict[j]=matched_base.count(j)/total_num_of_base\n","      if freq_dict[j]==0:\n","        shanon_dict[j]=0\n","      else:\n","        shanon_dict[j]=freq_dict[j]*np.log(freq_dict[j])\n","    shanon_ent.append(-sum(shanon_dict.values()))\n","  else:\n","    shanon_ent.append(0)\n","\n","pos_p_1=pileup['pos']+1\n","rst_bedform=pd.concat([pileup[['chrom', 'pos']],pos_p_1,pd.DataFrame(shanon_ent)],axis=1)\n","rst_bedform.to_csv(\"bed_formf_1.csv\",sep=\" \",header=None,index=False)"],"execution_count":155,"outputs":[]},{"cell_type":"code","metadata":{"id":"N-B8vX_qBPoD"},"source":[""],"execution_count":null,"outputs":[]}]}