{"nbformat":4,"nbformat_minor":0,"metadata":{"colab":{"name":"CoLab_TermProj_M2.ipynb","provenance":[],"collapsed_sections":[],"toc_visible":true},"kernelspec":{"name":"python3","display_name":"Python 3"}},"cells":[{"cell_type":"markdown","metadata":{"id":"_SmTogjeMD47"},"source":["# 생물정보학 및 실습 1 - Term Project용 Mission 2 환경 셋업\n","\n","```\n","생물정보학 및 실습 1\n","서울대학교 협동과정 생물정보학전공 2021년 1학기\n","\n","이 노트북은 Creative Commons BY-SA 조건으로 재사용 가능합니다.\n","장혜식 <hyeshik@snu.ac.kr>, 서울대학교 생명과학부\n","```\n","\n","이 노트북은 Google Colab을 이용해서 Term Project를 할 수 있도록 환경을 준비해 줍니다. 터미널 사용이 원활하지 않은 상태이고, 데이터를 오랫동안 보관하기 어렵기 때문에 되도록 개인용 컴퓨터에서 [Anaconda](https://www.anaconda.com/)나 [JupyterLab](https://jupyterlab.readthedocs.io/en/stable/) 등을 설치하고 사용하시는 것이 편하지만, 개인용 컴퓨터 사양이 부족한 경우에는 Google CoLab을 사용할 수도 있습니다. 개인용 컴퓨터를 쓸 때는 Windows에서는 [WSL2](https://docs.microsoft.com/en-us/windows/wsl/install-win10)를 쓰시면 리눅스 커맨드라인 환경을 쓸 수 있습니다. Google Colab은 창을 닫은 후 90분 후, 또는 창을 처음 열고 12시간 후에 환경을 닫아버리기 때문에 매번 실습할 때마다 새로 셋업이 필요합니다. `Runtime->Run All`을 실행해서 전체를 한 번 실행하시면 준비됩니다. 모든 것이 잘 안 되고 뭔가 꼬인 느낌이 강하게 들 때는 `Runtime->Factory reset runtime`을 누르고 새로 시작하시면 될 수도 있습니다.\n"]},{"cell_type":"markdown","metadata":{"id":"zGYjheLcRJS0"},"source":["## Google Drive 마운트\n","\n","데이터 파일들을 불러오고 저장하기 위해서 Google Drive를 연결합니다. **이번엔 스크립트 파일을 편집해야 하기 때문에 Google Drive 마운트를 하지 않으면 vim등 콘솔에서 쓸 수 있는 에디터를 쓰셔야 합니다.** 화면에 나오는 주소를 눌러서 인증하고 인증 코드를 그 아래에 입력하세요. 그러면 `/contents/drive/MyDrive`에서 자신의 Drive 최상단 폴더가 보입니다."]},{"cell_type":"code","metadata":{"id":"0HgOxANiRHgl","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1622741761383,"user_tz":-540,"elapsed":15769,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"588b4ab1-5dd3-40c8-e322-19937908845d"},"source":["from google.colab import drive\n","drive.mount('/content/drive')"],"execution_count":1,"outputs":[{"output_type":"stream","text":["Mounted at /content/drive\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"OcEjkQFq69tz","executionInfo":{"status":"ok","timestamp":1622743432533,"user_tz":-540,"elapsed":489,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"6e43ccf7-1d6b-4fd6-ebb7-d5ccb5db09b1"},"source":["%cd '/content/drive/MyDrive/Colab_Notebooks'"],"execution_count":34,"outputs":[{"output_type":"stream","text":["/content/drive/MyDrive/Colab_Notebooks\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"ra3RB22DWEER"},"source":["## 실습용 Bioconda 환경 설치하기\n","실습용 프로그램들을 쉽게 설치할 수 있도록 [Bioconda](https://bioconda.github.io/) 환경을 만듭니다. 실행하고 진행상황을 관찰하다가 에러로 보이는 것이 있으면 잘 읽어보고 해결합니다.\n","\n","파이썬 모듈들은 Colab쪽의 파이썬 인터프리터를 바꿀 수 없기 때문에 Conda에 설치하더라도 사용하기 어렵습니다. 파이썬 모듈은 `pip` 명령어로 설치하시면 Colab에서 불러와서 사용할 수 있습니다.\n","\n","환경 초기화에 관련된 스크립트는 [GitHub 프로젝트](https://github.com/hyeshik/colab-biolab)에 있습니다."]},{"cell_type":"code","metadata":{"id":"TTXmNi4hcap3","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1622741884361,"user_tz":-540,"elapsed":118449,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"51797154-d08b-4533-80af-0c6c5fcd2442"},"source":["!git clone https://github.com/hyeshik/colab-biolab.git\n","!cd colab-biolab && bash tools/setup.sh\n","exec(open('colab-biolab/tools/activate_conda.py').read())"],"execution_count":3,"outputs":[{"output_type":"stream","text":["Cloning into 'colab-biolab'...\n","remote: Enumerating objects: 69, done.\u001b[K\n","remote: Counting objects: 100% (69/69), done.\u001b[K\n","remote: Compressing objects: 100% (43/43), done.\u001b[K\n","remote: Total 69 (delta 22), reused 54 (delta 13), pack-reused 0\u001b[K\n","Unpacking objects: 100% (69/69), done.\n","./\n","./root/\n","./root/.bashrc.biolab\n","./root/.bin.priority/\n","./root/.bin.priority/pip\n","./root/.bin.priority/pip2\n","./root/.bin.priority/pip3\n","./root/.condarc\n","./root/.profile\n","./root/.tmux.conf\n","./root/.vimrc\n","--2021-06-03 17:36:08--  https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh\n","Resolving repo.anaconda.com (repo.anaconda.com)... 104.16.131.3, 104.16.130.3, 2606:4700::6810:8303, ...\n","Connecting to repo.anaconda.com (repo.anaconda.com)|104.16.131.3|:443... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 94235922 (90M) [application/x-sh]\n","Saving to: ‘miniconda3.sh’\n","\n","miniconda3.sh       100%[===================>]  89.87M  40.0MB/s    in 2.2s    \n","\n","2021-06-03 17:36:10 (40.0 MB/s) - ‘miniconda3.sh’ saved [94235922/94235922]\n","\n","PREFIX=/root/conda\n","Unpacking payload ...\n","Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\bdone\n","Solving environment: - \b\b\\ \b\bdone\n","\n","## Package Plan ##\n","\n","  environment location: /root/conda\n","\n","  added / updated specs:\n","    - _libgcc_mutex==0.1=main\n","    - brotlipy==0.7.0=py38h27cfd23_1003\n","    - ca-certificates==2020.10.14=0\n","    - certifi==2020.6.20=pyhd3eb1b0_3\n","    - cffi==1.14.3=py38h261ae71_2\n","    - chardet==3.0.4=py38h06a4308_1003\n","    - conda-package-handling==1.7.2=py38h03888b9_0\n","    - conda==4.9.2=py38h06a4308_0\n","    - cryptography==3.2.1=py38h3c74f83_1\n","    - idna==2.10=py_0\n","    - ld_impl_linux-64==2.33.1=h53a641e_7\n","    - libedit==3.1.20191231=h14c3975_1\n","    - libffi==3.3=he6710b0_2\n","    - libgcc-ng==9.1.0=hdf63c60_0\n","    - libstdcxx-ng==9.1.0=hdf63c60_0\n","    - ncurses==6.2=he6710b0_1\n","    - openssl==1.1.1h=h7b6447c_0\n","    - pip==20.2.4=py38h06a4308_0\n","    - pycosat==0.6.3=py38h7b6447c_1\n","    - pycparser==2.20=py_2\n","    - pyopenssl==19.1.0=pyhd3eb1b0_1\n","    - pysocks==1.7.1=py38h06a4308_0\n","    - python==3.8.5=h7579374_1\n","    - readline==8.0=h7b6447c_0\n","    - requests==2.24.0=py_0\n","    - ruamel_yaml==0.15.87=py38h7b6447c_1\n","    - setuptools==50.3.1=py38h06a4308_1\n","    - six==1.15.0=py38h06a4308_0\n","    - sqlite==3.33.0=h62c20be_0\n","    - tk==8.6.10=hbc83047_0\n","    - tqdm==4.51.0=pyhd3eb1b0_0\n","    - urllib3==1.25.11=py_0\n","    - wheel==0.35.1=pyhd3eb1b0_0\n","    - xz==5.2.5=h7b6447c_0\n","    - yaml==0.2.5=h7b6447c_0\n","    - zlib==1.2.11=h7b6447c_3\n","\n","\n","The following NEW packages will be INSTALLED:\n","\n","  _libgcc_mutex      pkgs/main/linux-64::_libgcc_mutex-0.1-main\n","  brotlipy           pkgs/main/linux-64::brotlipy-0.7.0-py38h27cfd23_1003\n","  ca-certificates    pkgs/main/linux-64::ca-certificates-2020.10.14-0\n","  certifi            pkgs/main/noarch::certifi-2020.6.20-pyhd3eb1b0_3\n","  cffi               pkgs/main/linux-64::cffi-1.14.3-py38h261ae71_2\n","  chardet            pkgs/main/linux-64::chardet-3.0.4-py38h06a4308_1003\n","  conda              pkgs/main/linux-64::conda-4.9.2-py38h06a4308_0\n","  conda-package-han~ pkgs/main/linux-64::conda-package-handling-1.7.2-py38h03888b9_0\n","  cryptography       pkgs/main/linux-64::cryptography-3.2.1-py38h3c74f83_1\n","  idna               pkgs/main/noarch::idna-2.10-py_0\n","  ld_impl_linux-64   pkgs/main/linux-64::ld_impl_linux-64-2.33.1-h53a641e_7\n","  libedit            pkgs/main/linux-64::libedit-3.1.20191231-h14c3975_1\n","  libffi             pkgs/main/linux-64::libffi-3.3-he6710b0_2\n","  libgcc-ng          pkgs/main/linux-64::libgcc-ng-9.1.0-hdf63c60_0\n","  libstdcxx-ng       pkgs/main/linux-64::libstdcxx-ng-9.1.0-hdf63c60_0\n","  ncurses            pkgs/main/linux-64::ncurses-6.2-he6710b0_1\n","  openssl            pkgs/main/linux-64::openssl-1.1.1h-h7b6447c_0\n","  pip                pkgs/main/linux-64::pip-20.2.4-py38h06a4308_0\n","  pycosat            pkgs/main/linux-64::pycosat-0.6.3-py38h7b6447c_1\n","  pycparser          pkgs/main/noarch::pycparser-2.20-py_2\n","  pyopenssl          pkgs/main/noarch::pyopenssl-19.1.0-pyhd3eb1b0_1\n","  pysocks            pkgs/main/linux-64::pysocks-1.7.1-py38h06a4308_0\n","  python             pkgs/main/linux-64::python-3.8.5-h7579374_1\n","  readline           pkgs/main/linux-64::readline-8.0-h7b6447c_0\n","  requests           pkgs/main/noarch::requests-2.24.0-py_0\n","  ruamel_yaml        pkgs/main/linux-64::ruamel_yaml-0.15.87-py38h7b6447c_1\n","  setuptools         pkgs/main/linux-64::setuptools-50.3.1-py38h06a4308_1\n","  six                pkgs/main/linux-64::six-1.15.0-py38h06a4308_0\n","  sqlite             pkgs/main/linux-64::sqlite-3.33.0-h62c20be_0\n","  tk                 pkgs/main/linux-64::tk-8.6.10-hbc83047_0\n","  tqdm               pkgs/main/noarch::tqdm-4.51.0-pyhd3eb1b0_0\n","  urllib3            pkgs/main/noarch::urllib3-1.25.11-py_0\n","  wheel              pkgs/main/noarch::wheel-0.35.1-pyhd3eb1b0_0\n","  xz                 pkgs/main/linux-64::xz-5.2.5-h7b6447c_0\n","  yaml               pkgs/main/linux-64::yaml-0.2.5-h7b6447c_0\n","  zlib               pkgs/main/linux-64::zlib-1.2.11-h7b6447c_3\n","\n","\n","Preparing transaction: / \b\b- \b\b\\ \b\b| \b\bdone\n","Executing transaction: - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n","installation finished.\n","WARNING:\n","    You currently have a PYTHONPATH environment variable set. This may cause\n","    unexpected behavior when running the Python interpreter in Miniconda3.\n","    For best results, please verify that your PYTHONPATH only points to\n","    directories of packages that are compatible with the Python interpreter\n","    in Miniconda3: /root/conda\n","Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n","Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n","\n","\n","==> WARNING: A newer version of conda exists. <==\n","  current version: 4.9.2\n","  latest version: 4.10.1\n","\n","Please update conda by running\n","\n","    $ conda update -n base -c defaults conda\n","\n","\n","\n","## Package Plan ##\n","\n","  environment location: /root/conda/envs/lab\n","\n","  added / updated specs:\n","    - bedtools\n","    - biopython\n","    - bwa\n","    - cython\n","    - matplotlib\n","    - numpy\n","    - pandas\n","    - pip\n","    - pysam\n","    - samtools\n","    - scipy\n","    - star\n","\n","\n","The following packages will be downloaded:\n","\n","    package                    |            build\n","    ---------------------------|-----------------\n","    _libgcc_mutex-0.1          |      conda_forge           3 KB  conda-forge\n","    _openmp_mutex-4.5          |            1_gnu          22 KB  conda-forge\n","    bedtools-2.30.0            |       h7d7f7ad_1        17.9 MB  bioconda\n","    biopython-1.78             |   py37h5e8e339_2         2.6 MB  conda-forge\n","    bwa-0.7.17                 |       h5bf99c6_8         608 KB  bioconda\n","    bzip2-1.0.8                |       h7f98852_4         484 KB  conda-forge\n","    c-ares-1.17.1              |       h7f98852_1         109 KB  conda-forge\n","    ca-certificates-2021.5.30  |       ha878542_0         136 KB  conda-forge\n","    certifi-2021.5.30          |   py37h89c1867_0         141 KB  conda-forge\n","    cycler-0.10.0              |             py_2           9 KB  conda-forge\n","    cython-0.29.23             |   py37hcd2ae1e_0         2.2 MB  conda-forge\n","    dbus-1.13.6                |       h48d8840_2         572 KB  conda-forge\n","    expat-2.4.1                |       h9c3ff4c_0         182 KB  conda-forge\n","    fontconfig-2.13.1          |    hba837de_1005         357 KB  conda-forge\n","    freetype-2.10.4            |       h0708190_1         890 KB  conda-forge\n","    gettext-0.19.8.1           |    h0b5b191_1005         3.6 MB  conda-forge\n","    glib-2.68.2                |       h9c3ff4c_2         450 KB  conda-forge\n","    glib-tools-2.68.2          |       h9c3ff4c_2          86 KB  conda-forge\n","    gst-plugins-base-1.14.0    |       hbbd80ab_1         4.8 MB\n","    gstreamer-1.14.0           |       h28cd5cc_2         3.2 MB\n","    htslib-1.12                |       h9093b5e_1         2.1 MB  bioconda\n","    icu-58.2                   |    hf484d3e_1000        22.6 MB  conda-forge\n","    jbig-2.1                   |    h7f98852_2003          43 KB  conda-forge\n","    jpeg-9d                    |       h36c2ea0_0         264 KB  conda-forge\n","    kiwisolver-1.3.1           |   py37h2527ec5_1          78 KB  conda-forge\n","    krb5-1.19.1                |       hcc1bbae_0         1.4 MB  conda-forge\n","    lcms2-2.12                 |       hddcbb42_0         443 KB  conda-forge\n","    ld_impl_linux-64-2.35.1    |       hea4e1c9_2         618 KB  conda-forge\n","    lerc-2.2.1                 |       h9c3ff4c_0         213 KB  conda-forge\n","    libblas-3.9.0              |       9_openblas          11 KB  conda-forge\n","    libcblas-3.9.0             |       9_openblas          11 KB  conda-forge\n","    libcurl-7.77.0             |       h2574ce0_0         334 KB  conda-forge\n","    libdeflate-1.7             |       h7f98852_5          67 KB  conda-forge\n","    libedit-3.1.20191231       |       he28a2e2_2         121 KB  conda-forge\n","    libev-4.33                 |       h516909a_1         104 KB  conda-forge\n","    libffi-3.3                 |       h58526e2_2          51 KB  conda-forge\n","    libgcc-ng-9.3.0            |      h2828fa1_19         7.8 MB  conda-forge\n","    libgfortran-ng-9.3.0       |      hff62375_19          22 KB  conda-forge\n","    libgfortran5-9.3.0         |      hff62375_19         2.0 MB  conda-forge\n","    libglib-2.68.2             |       h3e27bee_2         3.1 MB  conda-forge\n","    libgomp-9.3.0              |      h2828fa1_19         376 KB  conda-forge\n","    libiconv-1.16              |       h516909a_0         1.4 MB  conda-forge\n","    liblapack-3.9.0            |       9_openblas          11 KB  conda-forge\n","    libnghttp2-1.43.0          |       h812cca2_0         808 KB  conda-forge\n","    libopenblas-0.3.15         |pthreads_h8fe5266_1         9.2 MB  conda-forge\n","    libpng-1.6.37              |       h21135ba_2         306 KB  conda-forge\n","    libssh2-1.9.0              |       ha56f1ee_6         226 KB  conda-forge\n","    libstdcxx-ng-9.3.0         |      h6de172a_19         4.0 MB  conda-forge\n","    libtiff-4.3.0              |       hf544144_1         668 KB  conda-forge\n","    libuuid-2.32.1             |    h7f98852_1000          28 KB  conda-forge\n","    libwebp-base-1.2.0         |       h7f98852_2         815 KB  conda-forge\n","    libxcb-1.13                |    h7f98852_1003         395 KB  conda-forge\n","    libxml2-2.9.10             |       hb55368b_3         1.2 MB\n","    lz4-c-1.9.3                |       h9c3ff4c_0         179 KB  conda-forge\n","    matplotlib-3.4.2           |   py37h89c1867_0           7 KB  conda-forge\n","    matplotlib-base-3.4.2      |   py37hdd32ed1_0         7.2 MB  conda-forge\n","    ncurses-6.2                |       h58526e2_4         985 KB  conda-forge\n","    numpy-1.20.3               |   py37h038b26d_1         5.7 MB  conda-forge\n","    olefile-0.46               |     pyh9f0ad1d_1          32 KB  conda-forge\n","    openjpeg-2.4.0             |       hb52868f_1         444 KB  conda-forge\n","    openssl-1.1.1k             |       h7f98852_0         2.1 MB  conda-forge\n","    pandas-1.2.4               |   py37h219a48f_0        11.8 MB  conda-forge\n","    pcre-8.44                  |       he1b5a44_0         261 KB  conda-forge\n","    perl-5.32.0                |       h36c2ea0_0        16.8 MB  conda-forge\n","    pillow-8.2.0               |   py37h4600e1f_1         684 KB  conda-forge\n","    pip-21.1.2                 |     pyhd8ed1ab_0         1.1 MB  conda-forge\n","    pthread-stubs-0.4          |    h36c2ea0_1001           5 KB  conda-forge\n","    pyparsing-2.4.7            |     pyh9f0ad1d_0          60 KB  conda-forge\n","    pyqt-5.9.2                 |   py37hcca6a23_4         5.7 MB  conda-forge\n","    pysam-0.16.0.1             |   py37h45aed0b_3         2.6 MB  bioconda\n","    python-3.7.10              |hffdb5ce_100_cpython        57.3 MB  conda-forge\n","    python-dateutil-2.8.1      |             py_0         220 KB  conda-forge\n","    python_abi-3.7             |          1_cp37m           4 KB  conda-forge\n","    pytz-2021.1                |     pyhd8ed1ab_0         239 KB  conda-forge\n","    qt-5.9.7                   |       h5867ecd_1        68.5 MB\n","    readline-8.1               |       h46c0cb4_0         295 KB  conda-forge\n","    samtools-1.12              |       h9aed4be_1         386 KB  bioconda\n","    scipy-1.6.3                |   py37h29e03ee_0        20.5 MB  conda-forge\n","    setuptools-49.6.0          |   py37h89c1867_3         947 KB  conda-forge\n","    sip-4.19.8                 |   py37hf484d3e_0         274 KB\n","    six-1.16.0                 |     pyh6c4a22f_0          14 KB  conda-forge\n","    sqlite-3.35.5              |       h74cdb3f_0         1.4 MB  conda-forge\n","    star-2.7.9a                |       h9ee0642_0         3.8 MB  bioconda\n","    tk-8.6.10                  |       h21135ba_1         3.2 MB  conda-forge\n","    tornado-6.1                |   py37h5e8e339_1         646 KB  conda-forge\n","    wheel-0.36.2               |     pyhd3deb0d_0          31 KB  conda-forge\n","    xorg-libxau-1.0.9          |       h7f98852_0          13 KB  conda-forge\n","    xorg-libxdmcp-1.1.3        |       h7f98852_0          19 KB  conda-forge\n","    xz-5.2.5                   |       h516909a_1         343 KB  conda-forge\n","    zlib-1.2.11                |    h516909a_1010         106 KB  conda-forge\n","    zstd-1.5.0                 |       ha95c52a_0         490 KB  conda-forge\n","    ------------------------------------------------------------\n","                                           Total:       313.2 MB\n","\n","The following NEW packages will be INSTALLED:\n","\n","  _libgcc_mutex      conda-forge/linux-64::_libgcc_mutex-0.1-conda_forge\n","  _openmp_mutex      conda-forge/linux-64::_openmp_mutex-4.5-1_gnu\n","  bedtools           bioconda/linux-64::bedtools-2.30.0-h7d7f7ad_1\n","  biopython          conda-forge/linux-64::biopython-1.78-py37h5e8e339_2\n","  bwa                bioconda/linux-64::bwa-0.7.17-h5bf99c6_8\n","  bzip2              conda-forge/linux-64::bzip2-1.0.8-h7f98852_4\n","  c-ares             conda-forge/linux-64::c-ares-1.17.1-h7f98852_1\n","  ca-certificates    conda-forge/linux-64::ca-certificates-2021.5.30-ha878542_0\n","  certifi            conda-forge/linux-64::certifi-2021.5.30-py37h89c1867_0\n","  cycler             conda-forge/noarch::cycler-0.10.0-py_2\n","  cython             conda-forge/linux-64::cython-0.29.23-py37hcd2ae1e_0\n","  dbus               conda-forge/linux-64::dbus-1.13.6-h48d8840_2\n","  expat              conda-forge/linux-64::expat-2.4.1-h9c3ff4c_0\n","  fontconfig         conda-forge/linux-64::fontconfig-2.13.1-hba837de_1005\n","  freetype           conda-forge/linux-64::freetype-2.10.4-h0708190_1\n","  gettext            conda-forge/linux-64::gettext-0.19.8.1-h0b5b191_1005\n","  glib               conda-forge/linux-64::glib-2.68.2-h9c3ff4c_2\n","  glib-tools         conda-forge/linux-64::glib-tools-2.68.2-h9c3ff4c_2\n","  gst-plugins-base   pkgs/main/linux-64::gst-plugins-base-1.14.0-hbbd80ab_1\n","  gstreamer          pkgs/main/linux-64::gstreamer-1.14.0-h28cd5cc_2\n","  htslib             bioconda/linux-64::htslib-1.12-h9093b5e_1\n","  icu                conda-forge/linux-64::icu-58.2-hf484d3e_1000\n","  jbig               conda-forge/linux-64::jbig-2.1-h7f98852_2003\n","  jpeg               conda-forge/linux-64::jpeg-9d-h36c2ea0_0\n","  kiwisolver         conda-forge/linux-64::kiwisolver-1.3.1-py37h2527ec5_1\n","  krb5               conda-forge/linux-64::krb5-1.19.1-hcc1bbae_0\n","  lcms2              conda-forge/linux-64::lcms2-2.12-hddcbb42_0\n","  ld_impl_linux-64   conda-forge/linux-64::ld_impl_linux-64-2.35.1-hea4e1c9_2\n","  lerc               conda-forge/linux-64::lerc-2.2.1-h9c3ff4c_0\n","  libblas            conda-forge/linux-64::libblas-3.9.0-9_openblas\n","  libcblas           conda-forge/linux-64::libcblas-3.9.0-9_openblas\n","  libcurl            conda-forge/linux-64::libcurl-7.77.0-h2574ce0_0\n","  libdeflate         conda-forge/linux-64::libdeflate-1.7-h7f98852_5\n","  libedit            conda-forge/linux-64::libedit-3.1.20191231-he28a2e2_2\n","  libev              conda-forge/linux-64::libev-4.33-h516909a_1\n","  libffi             conda-forge/linux-64::libffi-3.3-h58526e2_2\n","  libgcc-ng          conda-forge/linux-64::libgcc-ng-9.3.0-h2828fa1_19\n","  libgfortran-ng     conda-forge/linux-64::libgfortran-ng-9.3.0-hff62375_19\n","  libgfortran5       conda-forge/linux-64::libgfortran5-9.3.0-hff62375_19\n","  libglib            conda-forge/linux-64::libglib-2.68.2-h3e27bee_2\n","  libgomp            conda-forge/linux-64::libgomp-9.3.0-h2828fa1_19\n","  libiconv           conda-forge/linux-64::libiconv-1.16-h516909a_0\n","  liblapack          conda-forge/linux-64::liblapack-3.9.0-9_openblas\n","  libnghttp2         conda-forge/linux-64::libnghttp2-1.43.0-h812cca2_0\n","  libopenblas        conda-forge/linux-64::libopenblas-0.3.15-pthreads_h8fe5266_1\n","  libpng             conda-forge/linux-64::libpng-1.6.37-h21135ba_2\n","  libssh2            conda-forge/linux-64::libssh2-1.9.0-ha56f1ee_6\n","  libstdcxx-ng       conda-forge/linux-64::libstdcxx-ng-9.3.0-h6de172a_19\n","  libtiff            conda-forge/linux-64::libtiff-4.3.0-hf544144_1\n","  libuuid            conda-forge/linux-64::libuuid-2.32.1-h7f98852_1000\n","  libwebp-base       conda-forge/linux-64::libwebp-base-1.2.0-h7f98852_2\n","  libxcb             conda-forge/linux-64::libxcb-1.13-h7f98852_1003\n","  libxml2            pkgs/main/linux-64::libxml2-2.9.10-hb55368b_3\n","  lz4-c              conda-forge/linux-64::lz4-c-1.9.3-h9c3ff4c_0\n","  matplotlib         conda-forge/linux-64::matplotlib-3.4.2-py37h89c1867_0\n","  matplotlib-base    conda-forge/linux-64::matplotlib-base-3.4.2-py37hdd32ed1_0\n","  ncurses            conda-forge/linux-64::ncurses-6.2-h58526e2_4\n","  numpy              conda-forge/linux-64::numpy-1.20.3-py37h038b26d_1\n","  olefile            conda-forge/noarch::olefile-0.46-pyh9f0ad1d_1\n","  openjpeg           conda-forge/linux-64::openjpeg-2.4.0-hb52868f_1\n","  openssl            conda-forge/linux-64::openssl-1.1.1k-h7f98852_0\n","  pandas             conda-forge/linux-64::pandas-1.2.4-py37h219a48f_0\n","  pcre               conda-forge/linux-64::pcre-8.44-he1b5a44_0\n","  perl               conda-forge/linux-64::perl-5.32.0-h36c2ea0_0\n","  pillow             conda-forge/linux-64::pillow-8.2.0-py37h4600e1f_1\n","  pip                conda-forge/noarch::pip-21.1.2-pyhd8ed1ab_0\n","  pthread-stubs      conda-forge/linux-64::pthread-stubs-0.4-h36c2ea0_1001\n","  pyparsing          conda-forge/noarch::pyparsing-2.4.7-pyh9f0ad1d_0\n","  pyqt               conda-forge/linux-64::pyqt-5.9.2-py37hcca6a23_4\n","  pysam              bioconda/linux-64::pysam-0.16.0.1-py37h45aed0b_3\n","  python             conda-forge/linux-64::python-3.7.10-hffdb5ce_100_cpython\n","  python-dateutil    conda-forge/noarch::python-dateutil-2.8.1-py_0\n","  python_abi         conda-forge/linux-64::python_abi-3.7-1_cp37m\n","  pytz               conda-forge/noarch::pytz-2021.1-pyhd8ed1ab_0\n","  qt                 pkgs/main/linux-64::qt-5.9.7-h5867ecd_1\n","  readline           conda-forge/linux-64::readline-8.1-h46c0cb4_0\n","  samtools           bioconda/linux-64::samtools-1.12-h9aed4be_1\n","  scipy              conda-forge/linux-64::scipy-1.6.3-py37h29e03ee_0\n","  setuptools         conda-forge/linux-64::setuptools-49.6.0-py37h89c1867_3\n","  sip                pkgs/main/linux-64::sip-4.19.8-py37hf484d3e_0\n","  six                conda-forge/noarch::six-1.16.0-pyh6c4a22f_0\n","  sqlite             conda-forge/linux-64::sqlite-3.35.5-h74cdb3f_0\n","  star               bioconda/linux-64::star-2.7.9a-h9ee0642_0\n","  tk                 conda-forge/linux-64::tk-8.6.10-h21135ba_1\n","  tornado            conda-forge/linux-64::tornado-6.1-py37h5e8e339_1\n","  wheel              conda-forge/noarch::wheel-0.36.2-pyhd3deb0d_0\n","  xorg-libxau        conda-forge/linux-64::xorg-libxau-1.0.9-h7f98852_0\n","  xorg-libxdmcp      conda-forge/linux-64::xorg-libxdmcp-1.1.3-h7f98852_0\n","  xz                 conda-forge/linux-64::xz-5.2.5-h516909a_1\n","  zlib               conda-forge/linux-64::zlib-1.2.11-h516909a_1010\n","  zstd               conda-forge/linux-64::zstd-1.5.0-ha95c52a_0\n","\n","\n","\n","Downloading and Extracting Packages\n","libdeflate-1.7       | 67 KB     | : 100% 1.0/1 [00:00<00:00,  8.99it/s]\n","olefile-0.46         | 32 KB     | : 100% 1.0/1 [00:00<00:00, 20.54it/s]\n","ld_impl_linux-64-2.3 | 618 KB    | : 100% 1.0/1 [00:00<00:00,  4.50it/s]\n","pandas-1.2.4         | 11.8 MB   | : 100% 1.0/1 [00:03<00:00,  3.50s/it]\n","wheel-0.36.2         | 31 KB     | : 100% 1.0/1 [00:00<00:00, 22.14it/s]\n","libwebp-base-1.2.0   | 815 KB    | : 100% 1.0/1 [00:00<00:00,  4.51it/s]\n","ncurses-6.2          | 985 KB    | : 100% 1.0/1 [00:00<00:00,  1.80it/s]\n","freetype-2.10.4      | 890 KB    | : 100% 1.0/1 [00:00<00:00,  3.61it/s]\n","icu-58.2             | 22.6 MB   | : 100% 1.0/1 [00:04<00:00,  4.51s/it]\n","libcblas-3.9.0       | 11 KB     | : 100% 1.0/1 [00:00<00:00, 25.42it/s]\n","libxcb-1.13          | 395 KB    | : 100% 1.0/1 [00:00<00:00,  5.27it/s]\n","readline-8.1         | 295 KB    | : 100% 1.0/1 [00:00<00:00,  8.64it/s]\n","libedit-3.1.20191231 | 121 KB    | : 100% 1.0/1 [00:00<00:00, 17.87it/s]\n","fontconfig-2.13.1    | 357 KB    | : 100% 1.0/1 [00:00<00:00,  7.52it/s]\n","libglib-2.68.2       | 3.1 MB    | : 100% 1.0/1 [00:00<00:00,  1.18it/s]\n","bzip2-1.0.8          | 484 KB    | : 100% 1.0/1 [00:00<00:00,  6.80it/s]\n","jpeg-9d              | 264 KB    | : 100% 1.0/1 [00:00<00:00,  9.55it/s]\n","cycler-0.10.0        | 9 KB      | : 100% 1.0/1 [00:00<00:00, 29.52it/s]\n","kiwisolver-1.3.1     | 78 KB     | : 100% 1.0/1 [00:00<00:00, 20.52it/s]\n","bedtools-2.30.0      | 17.9 MB   | : 100% 1.0/1 [00:03<00:00,  3.76s/it]               \n","scipy-1.6.3          | 20.5 MB   | : 100% 1.0/1 [00:04<00:00,  4.57s/it]               \n","glib-2.68.2          | 450 KB    | : 100% 1.0/1 [00:00<00:00,  4.94it/s]\n","pyqt-5.9.2           | 5.7 MB    | : 100% 1.0/1 [00:01<00:00,  1.89s/it]\n","libuuid-2.32.1       | 28 KB     | : 100% 1.0/1 [00:00<00:00, 27.55it/s]\n","python-3.7.10        | 57.3 MB   | : 100% 1.0/1 [00:10<00:00, 10.10s/it]               \n","pytz-2021.1          | 239 KB    | : 100% 1.0/1 [00:00<00:00,  6.27it/s]\n","glib-tools-2.68.2    | 86 KB     | : 100% 1.0/1 [00:00<00:00, 14.70it/s]\n","libstdcxx-ng-9.3.0   | 4.0 MB    | : 100% 1.0/1 [00:00<00:00,  1.21it/s]\n","libev-4.33           | 104 KB    | : 100% 1.0/1 [00:00<00:00, 18.35it/s]\n","numpy-1.20.3         | 5.7 MB    | : 100% 1.0/1 [00:01<00:00,  1.71s/it]\n","libcurl-7.77.0       | 334 KB    | : 100% 1.0/1 [00:00<00:00, 10.76it/s]\n","libpng-1.6.37        | 306 KB    | : 100% 1.0/1 [00:00<00:00,  8.76it/s]\n","matplotlib-3.4.2     | 7 KB      | : 100% 1.0/1 [00:00<00:00, 27.86it/s]\n","gettext-0.19.8.1     | 3.6 MB    | : 100% 1.0/1 [00:01<00:00,  1.13s/it]\n","libnghttp2-1.43.0    | 808 KB    | : 100% 1.0/1 [00:00<00:00,  4.96it/s]\n","libgfortran-ng-9.3.0 | 22 KB     | : 100% 1.0/1 [00:00<00:00, 22.42it/s]\n","pthread-stubs-0.4    | 5 KB      | : 100% 1.0/1 [00:00<00:00, 34.78it/s]\n","matplotlib-base-3.4. | 7.2 MB    | : 100% 1.0/1 [00:01<00:00,  1.66s/it]               \n","cython-0.29.23       | 2.2 MB    | : 100% 1.0/1 [00:00<00:00,  1.51it/s]\n","xorg-libxdmcp-1.1.3  | 19 KB     | : 100% 1.0/1 [00:00<00:00, 23.10it/s]\n","python_abi-3.7       | 4 KB      | : 100% 1.0/1 [00:00<00:00, 28.88it/s]\n","six-1.16.0           | 14 KB     | : 100% 1.0/1 [00:00<00:00, 11.97it/s]\n","libgcc-ng-9.3.0      | 7.8 MB    | : 100% 1.0/1 [00:01<00:00,  1.59s/it]\n","libiconv-1.16        | 1.4 MB    | : 100% 1.0/1 [00:00<00:00,  3.99it/s]\n","qt-5.9.7             | 68.5 MB   | : 100% 1.0/1 [00:03<00:00,  3.25s/it]               \n","c-ares-1.17.1        | 109 KB    | : 100% 1.0/1 [00:00<00:00, 15.78it/s]\n","expat-2.4.1          | 182 KB    | : 100% 1.0/1 [00:00<00:00, 12.46it/s]\n","openjpeg-2.4.0       | 444 KB    | : 100% 1.0/1 [00:00<00:00,  7.76it/s]\n","gst-plugins-base-1.1 | 4.8 MB    | : 100% 1.0/1 [00:00<00:00,  4.27it/s]\n","samtools-1.12        | 386 KB    | : 100% 1.0/1 [00:00<00:00,  8.71it/s]\n","pyparsing-2.4.7      | 60 KB     | : 100% 1.0/1 [00:00<00:00, 20.47it/s]\n","ca-certificates-2021 | 136 KB    | : 100% 1.0/1 [00:00<00:00, 19.15it/s]\n","zstd-1.5.0           | 490 KB    | : 100% 1.0/1 [00:00<00:00,  7.88it/s]\n","pcre-8.44            | 261 KB    | : 100% 1.0/1 [00:00<00:00,  9.41it/s]\n","gstreamer-1.14.0     | 3.2 MB    | : 100% 1.0/1 [00:00<00:00,  2.74it/s]\n","htslib-1.12          | 2.1 MB    | : 100% 1.0/1 [00:00<00:00,  2.09it/s]\n","jbig-2.1             | 43 KB     | : 100% 1.0/1 [00:00<00:00, 25.14it/s]\n","xorg-libxau-1.0.9    | 13 KB     | : 100% 1.0/1 [00:00<00:00, 29.32it/s]\n","python-dateutil-2.8. | 220 KB    | : 100% 1.0/1 [00:00<00:00, 14.36it/s]\n","certifi-2021.5.30    | 141 KB    | : 100% 1.0/1 [00:00<00:00, 17.42it/s]\n","sqlite-3.35.5        | 1.4 MB    | : 100% 1.0/1 [00:00<00:00,  3.07it/s]\n","libssh2-1.9.0        | 226 KB    | : 100% 1.0/1 [00:00<00:00,  9.83it/s]\n","pillow-8.2.0         | 684 KB    | : 100% 1.0/1 [00:00<00:00,  4.95it/s]\n","openssl-1.1.1k       | 2.1 MB    | : 100% 1.0/1 [00:00<00:00,  2.39it/s]\n","lerc-2.2.1           | 213 KB    | : 100% 1.0/1 [00:00<00:00, 14.00it/s]\n","bwa-0.7.17           | 608 KB    | : 100% 1.0/1 [00:00<00:00,  7.87it/s]\n","perl-5.32.0          | 16.8 MB   | : 100% 1.0/1 [00:05<00:00,  5.13s/it]\n","star-2.7.9a          | 3.8 MB    | : 100% 1.0/1 [00:00<00:00,  1.30it/s]\n","setuptools-49.6.0    | 947 KB    | : 100% 1.0/1 [00:00<00:00,  2.96it/s]\n","pip-21.1.2           | 1.1 MB    | : 100% 1.0/1 [00:00<00:00,  2.50it/s]\n","libblas-3.9.0        | 11 KB     | : 100% 1.0/1 [00:00<00:00, 17.50it/s]\n","pysam-0.16.0.1       | 2.6 MB    | : 100% 1.0/1 [00:00<00:00,  1.73it/s]\n","libgfortran5-9.3.0   | 2.0 MB    | : 100% 1.0/1 [00:00<00:00,  1.87it/s]\n","liblapack-3.9.0      | 11 KB     | : 100% 1.0/1 [00:00<00:00, 31.98it/s]\n","libopenblas-0.3.15   | 9.2 MB    | : 100% 1.0/1 [00:02<00:00,  2.09s/it]\n","dbus-1.13.6          | 572 KB    | : 100% 1.0/1 [00:00<00:00,  5.96it/s]\n","tornado-6.1          | 646 KB    | : 100% 1.0/1 [00:00<00:00,  4.32it/s]\n","_libgcc_mutex-0.1    | 3 KB      | : 100% 1.0/1 [00:00<00:00, 31.25it/s]\n","sip-4.19.8           | 274 KB    | : 100% 1.0/1 [00:00<00:00, 10.79it/s]\n","tk-8.6.10            | 3.2 MB    | : 100% 1.0/1 [00:00<00:00,  1.34it/s]\n","libffi-3.3           | 51 KB     | : 100% 1.0/1 [00:00<00:00, 21.95it/s]\n","biopython-1.78       | 2.6 MB    | : 100% 1.0/1 [00:00<00:00,  1.04it/s]\n","lcms2-2.12           | 443 KB    | : 100% 1.0/1 [00:00<00:00,  8.03it/s]\n","lz4-c-1.9.3          | 179 KB    | : 100% 1.0/1 [00:00<00:00, 13.97it/s]\n","xz-5.2.5             | 343 KB    | : 100% 1.0/1 [00:00<00:00,  7.73it/s]\n","libgomp-9.3.0        | 376 KB    | : 100% 1.0/1 [00:00<00:00,  8.00it/s]\n","_openmp_mutex-4.5    | 22 KB     | : 100% 1.0/1 [00:00<00:00, 27.15it/s]\n","krb5-1.19.1          | 1.4 MB    | : 100% 1.0/1 [00:00<00:00,  2.66it/s]\n","zlib-1.2.11          | 106 KB    | : 100% 1.0/1 [00:00<00:00, 15.92it/s]\n","libtiff-4.3.0        | 668 KB    | : 100% 1.0/1 [00:00<00:00,  5.24it/s]\n","libxml2-2.9.10       | 1.2 MB    | : 100% 1.0/1 [00:00<00:00,  7.48it/s]\n","Preparing transaction: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n","Verifying transaction: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n","Executing transaction: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n","#\n","# To activate this environment, use\n","#\n","#     $ conda activate lab\n","#\n","# To deactivate an active environment, use\n","#\n","#     $ conda deactivate\n","\n","Activated conda environment `lab'!\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"eDfJ0fCzsTTb"},"source":["설정이 끝났으면, 이제 콘솔을 열어서 터미널로 접속할 수 있는 통로를 열어줍니다."]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"zvFnIOhF7Niy","executionInfo":{"status":"ok","timestamp":1622741888536,"user_tz":-540,"elapsed":1323,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"abfb82e3-50ca-4b70-fd8f-fb4d6f90d5f7"},"source":["!git init"],"execution_count":4,"outputs":[{"output_type":"stream","text":["Reinitialized existing Git repository in /content/drive/MyDrive/Colab Notebooks/.git/\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"Oj9mH1045wdy","executionInfo":{"status":"ok","timestamp":1622741895435,"user_tz":-540,"elapsed":830,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}}},"source":["!git config --global user.email \"0105jhjh@snu.ac.kr\""],"execution_count":5,"outputs":[]},{"cell_type":"code","metadata":{"id":"1rJ_fMqa7V8d","executionInfo":{"status":"ok","timestamp":1622741897067,"user_tz":-540,"elapsed":410,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}}},"source":["!git config --global user.name \"JaehongYu\""],"execution_count":6,"outputs":[]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"CFqK-rKp7dEM","executionInfo":{"status":"ok","timestamp":1622742388300,"user_tz":-540,"elapsed":489621,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"423b5470-9a0a-41ac-c9ae-9c7f0d1a2d0b"},"source":["!git add ."],"execution_count":7,"outputs":[{"output_type":"stream","text":["fatal: unable to stat 'binfo1-datapack1/gencode.gtf.gz': No such file or directory\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"ENwYct1J9G6R","executionInfo":{"status":"ok","timestamp":1622742390840,"user_tz":-540,"elapsed":804,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"09671a83-8c72-47cd-ed1c-210feae0c3f3"},"source":["!git commit -m \"all about problem 2 except the directory problem1 which is for problem 1\""],"execution_count":8,"outputs":[{"output_type":"stream","text":["On branch master\n","\n","Initial commit\n","\n","Untracked files:\n","\t\u001b[31mCoLab_TermProj.ipynb\u001b[m\n","\t\u001b[31mCoLab_TermProj_M2.ipynb\u001b[m\n","\t\u001b[31mbinfo1-datapack1/\u001b[m\n","\t\u001b[31mcolab-biolab/\u001b[m\n","\n","nothing added to commit but untracked files present\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"J2LqGiP_N26s"},"source":["## Google Drive에 시퀀싱 데이터 다운로드\n","이제 실습 수업에서 필요한 시퀀싱 데이터를 다운로드 받아서 설치합시다. 드라이브 공간만 충분하면 데이터는 계속 저장돼 있기 때문에 한 번만 하시면 됩니다. Colab에서는 작업용 데이터를 Google Drive 폴더 아래의 `binfo-datapack1`에 저장합니다. 실습용 데이터는 `/content/drive/MyDrive/binfo1-datapack1`에 풀리게 됩니다. 첫 번째에는 wget이 접속오류를 내는 경우가 많으니, 접속에러가 발생하면 다시 한 번 실행해 보세요. 1G씩 끊어지지만 자동으로 이어받아집니다."]},{"cell_type":"code","metadata":{"id":"kIN484e7NSxR"},"source":["!wget -O - --no-check-certificate https://hyeshik.qbio.io/binfo/binfo1-datapack1.tar | tar -C /content/drive/MyDrive -xf -"],"execution_count":null,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"lnlM-S70C-tY"},"source":["받은 데이터의 MD5 체크섬을 확인합니다. 한참 걸립니다. 각 파일들의 원래 체크섬은 다음과 같습니다. 한 번 비교해 보세요.\n","\n","```\n","140aaf30bcb9276cc716f8699f04ddd6  CLIP-35L33G.bam\n","f1b3336ed7e2f97d562dcc71641251bd  CLIP-35L33G.bam.bai\n","328883a73d507eafbf5b60bd6b906201  RNA-control.bam\n","02073818e2f398a73c3b76e5169de1ca  RNA-control.bam.bai\n","b09550d09d6c2a4ce27f0226f426fdb1  RNA-siLin28a.bam\n","fef112c727244060ea62d3f2564a07f6  RNA-siLin28a.bam.bai\n","28bbd0c47d725669340c784f1b772c01  RNA-siLuc.bam\n","43590fdc4d81905c0432e0d1cb8cfd5b  RNA-siLuc.bam.bai\n","5c08a9297307bc83259e658c4474f0cc  RPF-siLin28a.bam\n","a1bb3e29be412dfd7fd8d16b1b1acc4c  RPF-siLin28a.bam.bai\n","f2eebf50943024d0116c9cd3e744c707  RPF-siLuc.bam\n","dc24f69e8f571fc8be30f28ce5b84fcd  RPF-siLuc.bam.bai\n","```"]},{"cell_type":"code","metadata":{"id":"g8lC_YJRBCPf","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1622743376168,"user_tz":-540,"elapsed":611,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"63f94f62-1618-4411-b9fc-03199bb1aab9"},"source":["!md5sum \"drive/MyDrive/Colab_Notebooks/binfo1-datapack1/*\""],"execution_count":33,"outputs":[{"output_type":"stream","text":["md5sum: 'drive/MyDrive/Colab Notebooks/binfo1-datapack1/*': No such file or directory\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"oxm3KTMdAiSG"},"source":[""]},{"cell_type":"markdown","metadata":{"id":"kiQ5y4dTD27M"},"source":["그리고, GENCODE Annotation 데이터도 미리 받아서 저장해 둡니다. 에러가 나면 다시 한 번 실행해 봅니다."]},{"cell_type":"code","metadata":{"id":"Efdd3x8hDMh4","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1622743489600,"user_tz":-540,"elapsed":6635,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"03e464db-fbee-4e22-e7b2-ed0d1a3660c5"},"source":["!wget --no-check-certificate -O /content/drive/MyDrive/Colab_Notebooks/binfo1-datapack1/gencode.gtf.gz http://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M27/gencode.vM27.annotation.gtf.gz"],"execution_count":35,"outputs":[{"output_type":"stream","text":["--2021-06-03 18:04:42--  http://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_mouse/release_M27/gencode.vM27.annotation.gtf.gz\n","Resolving ftp.ebi.ac.uk (ftp.ebi.ac.uk)... 193.62.193.138\n","Connecting to ftp.ebi.ac.uk (ftp.ebi.ac.uk)|193.62.193.138|:80... connected.\n","HTTP request sent, awaiting response... 200 OK\n","Length: 28359457 (27M) [application/octet-stream]\n","Saving to: ‘/content/drive/MyDrive/Colab_Notebooks/binfo1-datapack1/gencode.gtf.gz’\n","\n","/content/drive/MyDr 100%[===================>]  27.04M  4.28MB/s    in 6.4s    \n","\n","2021-06-03 18:04:49 (4.23 MB/s) - ‘/content/drive/MyDrive/Colab_Notebooks/binfo1-datapack1/gencode.gtf.gz’ saved [28359457/28359457]\n","\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"O3rM9ds6EMhB"},"source":["다루기 쉽도록 Annotation 압축을 풉니다."]},{"cell_type":"code","metadata":{"id":"KNyuSM6PEQvX","executionInfo":{"status":"ok","timestamp":1622743518966,"user_tz":-540,"elapsed":9978,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}}},"source":["!gunzip /content/drive/MyDrive/Colab_Notebooks/binfo1-datapack1/gencode.gtf.gz"],"execution_count":37,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"ecEA1PPQUlIK"},"source":["실습에 필요한 프로그램들 몇 가지를 추가로 설치합니다."]},{"cell_type":"code","metadata":{"id":"neteyWsWEKBy","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1622742415231,"user_tz":-540,"elapsed":21111,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"88806814-dbca-4cc2-d62b-f82877c1a060"},"source":["!conda install -y bedtools bioawk"],"execution_count":12,"outputs":[{"output_type":"stream","text":["Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n","Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n","\n","\n","==> WARNING: A newer version of conda exists. <==\n","  current version: 4.9.2\n","  latest version: 4.10.1\n","\n","Please update conda by running\n","\n","    $ conda update -n base -c defaults conda\n","\n","\n","\n","## Package Plan ##\n","\n","  environment location: /root/conda/envs/lab\n","\n","  added / updated specs:\n","    - bedtools\n","    - bioawk\n","\n","\n","The following packages will be downloaded:\n","\n","    package                    |            build\n","    ---------------------------|-----------------\n","    bioawk-1.0                 |       h5bf99c6_6         188 KB  bioconda\n","    ------------------------------------------------------------\n","                                           Total:         188 KB\n","\n","The following NEW packages will be INSTALLED:\n","\n","  bioawk             bioconda/linux-64::bioawk-1.0-h5bf99c6_6\n","\n","\n","\n","Downloading and Extracting Packages\n","bioawk-1.0           | 188 KB    | : 100% 1.0/1 [00:00<00:00,  6.14it/s]               \n","Preparing transaction: - \b\bdone\n","Verifying transaction: | \b\bdone\n","Executing transaction: - \b\bdone\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"Oe_54MZrEi4y"},"source":["## Mission 2\n","\n","이번 미션에서는 start codon 주변에서 ribosome-protected fragment의 분포를 한 번 살펴보겠습니다. 보통은 RPF의 5'-end 분포가 start codon 앞 15nt 정도에서부터 본격적으로 많이 나오고, 거기서부터 CDS를 걸쳐서 3nt 주기로 올록볼록한 패턴이 나타납니다.\n","\n","이 그림을 제대로 그리려면 genome coordinate를 start codon을 기준으로 변환해서 작업해야 하지만, 이번 미션에서는 난이도를 적절히 내리기 위해서 start codon이 포함된 exon에 존재하는 5' end만 추려서 작업하도록 합시다. (여유가 좀 있어서 난이도를 적당히 올리고 싶은 분들은 negative strand를 포함할 수도 있고, start codon과 다른 exon에 있는 경우를 포함할 수도 있습니다. 또한 APPRIS 등을 기반으로 해서 transcript 중에 가장 )"]},{"cell_type":"code","metadata":{"id":"IANz6DmLUkA6","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1622743520105,"user_tz":-540,"elapsed":2,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"64f4b406-7251-4a3f-bf47-6d9ec3deb52e"},"source":["%cd '/content/drive/MyDrive/Colab_Notebooks/binfo1-datapack1/'"],"execution_count":38,"outputs":[{"output_type":"stream","text":["/content/drive/MyDrive/Colab_Notebooks/binfo1-datapack1\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"68uh7WVkUjhr"},"source":["우선 annotation에서 start codon이 포함된 exon만 추립니다. Gene 한 개에 isoform이 여러 개 있는 경우 모두 쓰면 read가 너무 부풀려질 수 있기 때문에 우선 transcript support level이 1인 것만 추려서 사용합니다. 여전히 TSL 1만 쓰더라도 부풀려질 가능성은 있기 때문에 제대로 하자면 APPRIS나 RNA-seq 데이터 기준으로 하나만 선정하는 것이 좋습니다만, 일단 여기서는 간단히..\n","\n","그리고, - strand가 들어가면 좀 계산이 복잡해지니까 또 간단하게 하기 위해서 + strand만 고릅니다. 이번엔 일단 무조건 간단하게 해 봅시다~"]},{"cell_type":"code","metadata":{"id":"vvFgM8hJUzDx","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1622743529992,"user_tz":-540,"elapsed":3537,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"c4a5ac35-c697-4657-d7c1-1cea4d95e910"},"source":["!grep '\tstart_codon\t.*\t+\t.*transcript_support_level \"1\"' gencode.gtf | \\\n"," sed -e 's/\t[^\t]*transcript_id \"\\([^\"]*\\)\".*$/\t\\1/g' > gencode-start.gtf\n","!head gencode-start.gtf"],"execution_count":39,"outputs":[{"output_type":"stream","text":["chr1\tHAVANA\tstart_codon\t4878137\t4878139\t.\t+\t0\tENSMUST00000134384.8\n","chr1\tHAVANA\tstart_codon\t4878137\t4878139\t.\t+\t0\tENSMUST00000027036.11\n","chr1\tHAVANA\tstart_codon\t4878137\t4878139\t.\t+\t0\tENSMUST00000150971.8\n","chr1\tHAVANA\tstart_codon\t4928137\t4928139\t.\t+\t0\tENSMUST00000081551.14\n","chr1\tHAVANA\tstart_codon\t5154674\t5154676\t.\t+\t0\tENSMUST00000044369.13\n","chr1\tHAVANA\tstart_codon\t5659272\t5659274\t.\t+\t0\tENSMUST00000160777.8\n","chr1\tHAVANA\tstart_codon\t5659272\t5659274\t.\t+\t0\tENSMUST00000027038.11\n","chr1\tHAVANA\tstart_codon\t6300227\t6300229\t.\t+\t0\tENSMUST00000027040.13\n","chr1\tHAVANA\tstart_codon\t6429555\t6429557\t.\t+\t0\tENSMUST00000133144.4\n","chr1\tHAVANA\tstart_codon\t6839122\t6839124\t.\t+\t0\tENSMUST00000140079.8\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"iWvIeTYOWsR6"},"source":["start codon가 들어가 있는 exon들을 고르기 위해서 미리 + strand에 있는 exon를 추립니다.. start codon 쪽에서 TSL 기준으로 골랐기 때문에 여기서는 고르지 않아도 됩니다. 물론 좀 느려지기는 하지만..."]},{"cell_type":"code","metadata":{"id":"yCqK2wyYWitn","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1622743573746,"user_tz":-540,"elapsed":40858,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"790d3e51-4f37-4690-fbe4-a10f0d553b0f"},"source":["!grep '\texon\t.*\t+\t' gencode.gtf | \\\n"," sed -e 's/\t[^\t]*transcript_id \"\\([^\"]*\\)\".*$/\t\\1/g' > gencode-plusexon.gtf\n","!head gencode-plusexon.gtf"],"execution_count":40,"outputs":[{"output_type":"stream","text":["chr1\tHAVANA\texon\t3143476\t3144545\t.\t+\t.\tENSMUST00000193812.2\n","chr1\tENSEMBL\texon\t3172239\t3172348\t.\t+\t.\tENSMUST00000082908.3\n","chr1\tHAVANA\texon\t3322980\t3323459\t.\t+\t.\tENSMUST00000192857.2\n","chr1\tHAVANA\texon\t3536810\t3536910\t.\t+\t.\tENSMUST00000161581.2\n","chr1\tHAVANA\texon\t3583628\t3583776\t.\t+\t.\tENSMUST00000161581.2\n","chr1\tHAVANA\texon\t3602018\t3602943\t.\t+\t.\tENSMUST00000192183.2\n","chr1\tHAVANA\texon\t3750378\t3752011\t.\t+\t.\tENSMUST00000193244.2\n","chr1\tHAVANA\texon\t3822233\t3824583\t.\t+\t.\tENSMUST00000194454.2\n","chr1\tHAVANA\texon\t4566774\t4569601\t.\t+\t.\tENSMUST00000193450.2\n","chr1\tHAVANA\texon\t4567697\t4567877\t.\t+\t.\tENSMUST00000194935.2\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"suULe5k6Yelf"},"source":["그 다음엔 exon 리스트와 start codon 리스트를 겹치고, transcript가 같은 것만 살려둡니다. 기왕 좀 다루기 쉽도록 출력은 bed로 하면서 좌표계를 바꿉니다. 결과는 bedtools에서 빠르게 다룰 수 있게 sort해 둡니다."]},{"cell_type":"code","metadata":{"id":"Zsv2_w5OXU9c","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1622743575856,"user_tz":-540,"elapsed":970,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"32092e64-2d04-4c60-f674-94d08c6251df"},"source":["!bedtools intersect -a gencode-start.gtf -b gencode-plusexon.gtf -wa -wb | \\\n"," awk -F'\t' -v OFS='\t' '$9 == $18 { print $10, $13-1, $14, $18, $4-1, $16; }' | \\\n"," sort -k1,1 -k2,3n -k4,4 > gencode-exons-containing-startcodon.bed\n","!head gencode-exons-containing-startcodon.bed; tail gencode-exons-containing-startcodon.bed"],"execution_count":41,"outputs":[{"output_type":"stream","text":["chr1\t4878010\t4878205\tENSMUST00000134384.8\t4878136\t+\n","chr1\t4878045\t4878205\tENSMUST00000027036.11\t4878136\t+\n","chr1\t4878052\t4878205\tENSMUST00000150971.8\t4878136\t+\n","chr1\t4928036\t4928199\tENSMUST00000081551.14\t4928136\t+\n","chr1\t5154639\t5154786\tENSMUST00000044369.13\t5154673\t+\n","chr1\t5659227\t5659528\tENSMUST00000027038.11\t5659271\t+\n","chr1\t5659257\t5659528\tENSMUST00000160777.8\t5659271\t+\n","chr1\t6300182\t6300297\tENSMUST00000027040.13\t6300226\t+\n","chr1\t6429441\t6429738\tENSMUST00000133144.4\t6429554\t+\n","chr1\t6839121\t6839176\tENSMUST00000043578.13\t6839121\t+\n","chrY\t80939672\t80939804\tENSMUST00000185340.2\t80939686\t+\n","chrY\t81470683\t81470815\tENSMUST00000187135.2\t81470697\t+\n","chrY\t82237903\t82238035\tENSMUST00000185636.2\t82237917\t+\n","chrY\t83043623\t83043755\tENSMUST00000187165.2\t83043637\t+\n","chrY\t84109956\t84110088\tENSMUST00000185776.7\t84109970\t+\n","chrY\t84759361\t84759493\tENSMUST00000186110.2\t84759375\t+\n","chrY\t86074448\t86074580\tENSMUST00000188754.2\t86074462\t+\n","chrY\t87129500\t87129632\tENSMUST00000189543.7\t87129514\t+\n","chrY\t87563647\t87563779\tENSMUST00000186493.2\t87563661\t+\n","chrY\t88065992\t88066124\tENSMUST00000187146.2\t88066006\t+\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"KvPNCpbtaYnC"},"source":["컬럼 4번에 transcript id, 5번에 start codon 위치를 넣었는데, start codon이 같은 것들이 여러개 눈에 띕니다. 아주 찝찝하지만, 우선 이번은 간단하게 할 것이기 때문에 눈을 질끈 감고 넘어가 봅시다.\n","\n","이제 alignment를 준비합니다. + strand alignment에서 read length가 짧은 것도 미리 필터링 합니다. RPF는 보통 25nt 이상이기 때문에 그 이하는 거의 확실히 rRNA, tRNA 등에서 오염된 것들입니다. SAM 파일 전체를 필터링해서 다시 인코딩하기 때문에 시간이 좀 걸립니다."]},{"cell_type":"code","metadata":{"id":"ZZaOeHo1alCB","executionInfo":{"status":"ok","timestamp":1622743697268,"user_tz":-540,"elapsed":121414,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}}},"source":["!(samtools view -H RPF-siLuc.bam; \\\n","  samtools view -F20 RPF-siLuc.bam | \\\n","  bioawk -c sam '{ if (length($seq) >= 25) print $0; }') | \\\n"," samtools view -b -o filtered-RPF-siLuc.bam"],"execution_count":42,"outputs":[]},{"cell_type":"markdown","metadata":{"id":"U1JDNb8ndTdk"},"source":["큰 일을 했으니 용량을 한 번 봅니다."]},{"cell_type":"code","metadata":{"id":"5OrnDxegFAip","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1622743697269,"user_tz":-540,"elapsed":19,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"11c0a833-a3e2-42e5-831c-5dad8dd0ad66"},"source":["!ls -al *RNA-siLuc.bam"],"execution_count":43,"outputs":[{"output_type":"stream","text":["-rw------- 1 root root 981684502 Jun  3 17:01 RNA-siLuc.bam\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"pr9zpKFmdi2e"},"source":["자 이제 read의 5' end 위치를 셉니다. bedtools의 `genomecov`를 쓰면 아주 편리합니다."]},{"cell_type":"code","metadata":{"id":"IZjHau7zftoU","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1622743961212,"user_tz":-540,"elapsed":263951,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"aa842f94-689d-4c5c-ca68-643b0d1ed729"},"source":["!bedtools genomecov -ibam filtered-RPF-siLuc.bam -bg -5 > fivepcounts-RPF-siLuc.bed\n","!head fivepcounts-RPF-siLuc.bed"],"execution_count":44,"outputs":[{"output_type":"stream","text":["tcmalloc: large alloc 1561239552 bytes == 0x56421c944000 @  0x7f75c9851887 0x56421a25afea 0x56421a25f6fd 0x56421a26033f 0x56421a2588a9 0x56421a20c773 0x7f75c8e7bbf7 0x56421a214569\n","tcmalloc: large alloc 1454047232 bytes == 0x56421c944000 @  0x7f75c9851887 0x56421a25afea 0x56421a25f6fd 0x56421a26033f 0x56421a2588a9 0x56421a20c773 0x7f75c8e7bbf7 0x56421a214569\n","chr1\t3176534\t3176535\t1\n","chr1\t3196256\t3196257\t1\n","chr1\t3221526\t3221527\t1\n","chr1\t3221542\t3221543\t2\n","chr1\t3221545\t3221546\t3\n","chr1\t3221546\t3221547\t2\n","chr1\t3221548\t3221550\t1\n","chr1\t3221571\t3221572\t1\n","chr1\t3221897\t3221898\t1\n","chr1\t3221994\t3221995\t1\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"9p4Jqwd9B21W","executionInfo":{"status":"ok","timestamp":1622743961214,"user_tz":-540,"elapsed":21,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"a84d3f20-8477-4ba2-b2d8-a1a214c977a3"},"source":["!head fivepcounts-RPF-siLuc.bed"],"execution_count":45,"outputs":[{"output_type":"stream","text":["chr1\t3176534\t3176535\t1\n","chr1\t3196256\t3196257\t1\n","chr1\t3221526\t3221527\t1\n","chr1\t3221542\t3221543\t2\n","chr1\t3221545\t3221546\t3\n","chr1\t3221546\t3221547\t2\n","chr1\t3221548\t3221550\t1\n","chr1\t3221571\t3221572\t1\n","chr1\t3221897\t3221898\t1\n","chr1\t3221994\t3221995\t1\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"y_AHnHnZmYJu"},"source":["그렇게 match 된 것들 중에 앞에서 찾은 start codon을 포함한 exon과 겹치는 것만 골라냅니다."]},{"cell_type":"code","metadata":{"id":"k1cOiX2Hftlx","colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"status":"ok","timestamp":1622743964518,"user_tz":-540,"elapsed":2079,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"745349c7-f487-42d1-da90-d4ca97fa5404"},"source":["!bedtools intersect -a fivepcounts-RPF-siLuc.bed -b gencode-exons-containing-startcodon.bed \\\n"," -wa -wb -nonamecheck > fivepcounts-filtered-RPF-siLuc.txt\n","!head fivepcounts-filtered-RPF-siLuc.txt"],"execution_count":46,"outputs":[{"output_type":"stream","text":["chr1\t4878048\t4878049\t1\tchr1\t4878010\t4878205\tENSMUST00000134384.8\t4878136\t+\n","chr1\t4878048\t4878049\t1\tchr1\t4878045\t4878205\tENSMUST00000027036.11\t4878136\t+\n","chr1\t4878077\t4878078\t1\tchr1\t4878010\t4878205\tENSMUST00000134384.8\t4878136\t+\n","chr1\t4878077\t4878078\t1\tchr1\t4878045\t4878205\tENSMUST00000027036.11\t4878136\t+\n","chr1\t4878077\t4878078\t1\tchr1\t4878052\t4878205\tENSMUST00000150971.8\t4878136\t+\n","chr1\t4878101\t4878102\t4\tchr1\t4878010\t4878205\tENSMUST00000134384.8\t4878136\t+\n","chr1\t4878101\t4878102\t4\tchr1\t4878045\t4878205\tENSMUST00000027036.11\t4878136\t+\n","chr1\t4878101\t4878102\t4\tchr1\t4878052\t4878205\tENSMUST00000150971.8\t4878136\t+\n","chr1\t4878103\t4878104\t1\tchr1\t4878010\t4878205\tENSMUST00000134384.8\t4878136\t+\n","chr1\t4878103\t4878104\t1\tchr1\t4878045\t4878205\tENSMUST00000027036.11\t4878136\t+\n"],"name":"stdout"}]},{"cell_type":"markdown","metadata":{"id":"0OcjYZ9bnL_6"},"source":["모든 준비가 끝났습니다. 이제 Figure S5A 도전해 보세요!\n","\n","* stop codon은 빼고 start codon 만입니다.\n","* 논문에서는 처리가 어려운 부분까지 모두 포함됐고, read를 중첩해서 세지 않았기 때문에, 여기서 세는 것과는 수치는 다를 수 있습니다.\n","* `pandas`로 한꺼번에 데이터를 모두 불러들이면 메모리가 부족할 수도 있습니다. 그런 경우에는 chunk를 쓰거나, 줄 단위로 처리해 보세요."]},{"cell_type":"code","metadata":{"id":"aPPW73t4X3pp","executionInfo":{"status":"ok","timestamp":1622742945070,"user_tz":-540,"elapsed":431,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}}},"source":["import pandas as pd"],"execution_count":21,"outputs":[]},{"cell_type":"code","metadata":{"id":"FJt-SZBHGLBh","executionInfo":{"status":"ok","timestamp":1622743993532,"user_tz":-540,"elapsed":904,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}}},"source":["dfd = pd.read_csv(\"fivepcounts-filtered-RPF-siLuc.txt\",header=None,sep='\\t')"],"execution_count":47,"outputs":[]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":419},"id":"NsryUcuXGSrc","executionInfo":{"status":"ok","timestamp":1622743997137,"user_tz":-540,"elapsed":338,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"2e143af7-daa6-4a5d-86a9-3425a52c6f96"},"source":["dfd"],"execution_count":48,"outputs":[{"output_type":"execute_result","data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>0</th>\n","      <th>1</th>\n","      <th>2</th>\n","      <th>3</th>\n","      <th>4</th>\n","      <th>5</th>\n","      <th>6</th>\n","      <th>7</th>\n","      <th>8</th>\n","      <th>9</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>chr1</td>\n","      <td>4878048</td>\n","      <td>4878049</td>\n","      <td>1</td>\n","      <td>chr1</td>\n","      <td>4878010</td>\n","      <td>4878205</td>\n","      <td>ENSMUST00000134384.8</td>\n","      <td>4878136</td>\n","      <td>+</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>chr1</td>\n","      <td>4878048</td>\n","      <td>4878049</td>\n","      <td>1</td>\n","      <td>chr1</td>\n","      <td>4878045</td>\n","      <td>4878205</td>\n","      <td>ENSMUST00000027036.11</td>\n","      <td>4878136</td>\n","      <td>+</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>chr1</td>\n","      <td>4878077</td>\n","      <td>4878078</td>\n","      <td>1</td>\n","      <td>chr1</td>\n","      <td>4878010</td>\n","      <td>4878205</td>\n","      <td>ENSMUST00000134384.8</td>\n","      <td>4878136</td>\n","      <td>+</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>chr1</td>\n","      <td>4878077</td>\n","      <td>4878078</td>\n","      <td>1</td>\n","      <td>chr1</td>\n","      <td>4878045</td>\n","      <td>4878205</td>\n","      <td>ENSMUST00000027036.11</td>\n","      <td>4878136</td>\n","      <td>+</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>chr1</td>\n","      <td>4878077</td>\n","      <td>4878078</td>\n","      <td>1</td>\n","      <td>chr1</td>\n","      <td>4878052</td>\n","      <td>4878205</td>\n","      <td>ENSMUST00000150971.8</td>\n","      <td>4878136</td>\n","      <td>+</td>\n","    </tr>\n","    <tr>\n","      <th>...</th>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","      <td>...</td>\n","    </tr>\n","    <tr>\n","      <th>340494</th>\n","      <td>chrY</td>\n","      <td>87129502</td>\n","      <td>87129503</td>\n","      <td>1</td>\n","      <td>chrY</td>\n","      <td>87129500</td>\n","      <td>87129632</td>\n","      <td>ENSMUST00000189543.7</td>\n","      <td>87129514</td>\n","      <td>+</td>\n","    </tr>\n","    <tr>\n","      <th>340495</th>\n","      <td>chrY</td>\n","      <td>87129574</td>\n","      <td>87129575</td>\n","      <td>1</td>\n","      <td>chrY</td>\n","      <td>87129500</td>\n","      <td>87129632</td>\n","      <td>ENSMUST00000189543.7</td>\n","      <td>87129514</td>\n","      <td>+</td>\n","    </tr>\n","    <tr>\n","      <th>340496</th>\n","      <td>chrY</td>\n","      <td>87563721</td>\n","      <td>87563722</td>\n","      <td>1</td>\n","      <td>chrY</td>\n","      <td>87563647</td>\n","      <td>87563779</td>\n","      <td>ENSMUST00000186493.2</td>\n","      <td>87563661</td>\n","      <td>+</td>\n","    </tr>\n","    <tr>\n","      <th>340497</th>\n","      <td>chrY</td>\n","      <td>88065994</td>\n","      <td>88065995</td>\n","      <td>1</td>\n","      <td>chrY</td>\n","      <td>88065992</td>\n","      <td>88066124</td>\n","      <td>ENSMUST00000187146.2</td>\n","      <td>88066006</td>\n","      <td>+</td>\n","    </tr>\n","    <tr>\n","      <th>340498</th>\n","      <td>chrY</td>\n","      <td>88066066</td>\n","      <td>88066067</td>\n","      <td>1</td>\n","      <td>chrY</td>\n","      <td>88065992</td>\n","      <td>88066124</td>\n","      <td>ENSMUST00000187146.2</td>\n","      <td>88066006</td>\n","      <td>+</td>\n","    </tr>\n","  </tbody>\n","</table>\n","<p>340499 rows × 10 columns</p>\n","</div>"],"text/plain":["           0         1         2  ...                      7         8  9\n","0       chr1   4878048   4878049  ...   ENSMUST00000134384.8   4878136  +\n","1       chr1   4878048   4878049  ...  ENSMUST00000027036.11   4878136  +\n","2       chr1   4878077   4878078  ...   ENSMUST00000134384.8   4878136  +\n","3       chr1   4878077   4878078  ...  ENSMUST00000027036.11   4878136  +\n","4       chr1   4878077   4878078  ...   ENSMUST00000150971.8   4878136  +\n","...      ...       ...       ...  ...                    ...       ... ..\n","340494  chrY  87129502  87129503  ...   ENSMUST00000189543.7  87129514  +\n","340495  chrY  87129574  87129575  ...   ENSMUST00000189543.7  87129514  +\n","340496  chrY  87563721  87563722  ...   ENSMUST00000186493.2  87563661  +\n","340497  chrY  88065994  88065995  ...   ENSMUST00000187146.2  88066006  +\n","340498  chrY  88066066  88066067  ...   ENSMUST00000187146.2  88066006  +\n","\n","[340499 rows x 10 columns]"]},"metadata":{"tags":[]},"execution_count":48}]},{"cell_type":"code","metadata":{"id":"xnlBzz1NGTjx","executionInfo":{"status":"ok","timestamp":1622744086916,"user_tz":-540,"elapsed":500,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}}},"source":["import matplotlib.pyplot as plt\n","import numpy as np"],"execution_count":50,"outputs":[]},{"cell_type":"code","metadata":{"id":"-Et3QLxP15CE","executionInfo":{"status":"ok","timestamp":1622744749388,"user_tz":-540,"elapsed":547,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}}},"source":["x=dfd.iloc[:,1]-dfd.iloc[:,8]\n","y=dfd.iloc[:,3]\n","list_x=np.array(x)"],"execution_count":67,"outputs":[]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"ymLsWRvxFccZ","executionInfo":{"status":"ok","timestamp":1622744372630,"user_tz":-540,"elapsed":415,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"461bc9ed-9e31-4a27-a3a8-e241e0976670"},"source":["min(x)"],"execution_count":56,"outputs":[{"output_type":"execute_result","data":{"text/plain":["-7536"]},"metadata":{"tags":[]},"execution_count":56}]},{"cell_type":"code","metadata":{"id":"EbE2gRpWGh9C","executionInfo":{"status":"ok","timestamp":1622744987477,"user_tz":-540,"elapsed":322,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}}},"source":["xx=list(range(-50,51))\n","yy=list(range(-50,51))"],"execution_count":76,"outputs":[]},{"cell_type":"code","metadata":{"id":"fcrvZ3m5HAUD","executionInfo":{"status":"ok","timestamp":1622745025585,"user_tz":-540,"elapsed":511,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}}},"source":["for i in range(-50,51):\n","  yy[i]=np.sum(y.iloc[np.where(list_x == yy[i])[0]])"],"execution_count":78,"outputs":[]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":265},"id":"XZU80NJ4HNQH","executionInfo":{"status":"ok","timestamp":1622745176661,"user_tz":-540,"elapsed":996,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"08f4a7ab-b2ec-4d59-ee5c-e8d44e1b229f"},"source":["plt.bar(xx, yy)\n","plt.savefig('result.png')"],"execution_count":82,"outputs":[{"output_type":"display_data","data":{"image/png":"iVBORw0KGgoAAAANSUhEUgAAAYMAAAD4CAYAAAAO9oqkAAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADh0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uMy4yLjIsIGh0dHA6Ly9tYXRwbG90bGliLm9yZy+WH4yJAAATjklEQVR4nO3df4xd5X3n8fendkhRu8QmTL3IY+1YitXKyW4SMgJXWa26sGsMRDF/pIioir2sN9YqZJVKkVqT/oE2PySilUqDNkGygjd2la5j0UZYwdT1OlRV/zBhCAnUENazBBZbgKexge6iJiL97h/3cXPXzHju2HfunR/vl3R1z/k+z7nzHI99P/c859zjVBWSpOXtl4Y9AEnS8BkGkiTDQJJkGEiSMAwkScDKYQ/gYl111VU1NjY27GFI0qLxxBNP/G1VjUzXtmjDYGxsjImJiWEPQ5IWjSQvztTmNJEkqbcwSLIqyYNJfpTk2SS/meTKJEeSnGjPq1vfJLkvyWSSp5Jc0/U621v/E0m2d9U/lOTpts19SdL/XZUkzaTXI4OvAH9eVb8BvB94FtgFHK2qDcDRtg5wE7ChPXYC9wMkuRK4G7gOuBa4+1yAtD6f7Npuy6XtliRpLmYNgyTvAv4V8ABAVf2sql4DtgJ7W7e9wK1teSuwrzqOAauSXA3cCBypqjNVdRY4AmxpbVdU1bHq3BtjX9drSZIGoJcjg/XAFPDfkjyZ5OtJfgVYU1Uvtz6vAGva8lrgpa7tT7baheonp6m/TZKdSSaSTExNTfUwdElSL3oJg5XANcD9VfVB4P/yiykhANon+nm/411V7a6q8aoaHxmZ9uooSdJF6CUMTgInq+qxtv4gnXB4tU3x0J5Pt/ZTwLqu7Udb7UL10WnqkqQBmTUMquoV4KUkv95KNwDPAAeBc1cEbQceassHgW3tqqJNwOttOukwsDnJ6nbieDNwuLW9kWRTu4poW9drSZIGoNcvnf0n4JtJLgOeB+6gEyQHkuwAXgRua30PATcDk8CbrS9VdSbJF4DHW7/PV9WZtvwp4BvA5cAj7SFJGpAs1v/cZnx8vPwGssZ2PQzAC/fcMuSRSAtfkieqany6Nr+BLEkyDCRJhoEkCcNAkoRhIEnCMJAkYRhIkjAMJEkYBpIkDANJEoaBJAnDQJKEYSBJwjCQJGEYSJIwDCRJGAaSJAwDSRKGgSQJw0CShGEgScIwkCRhGEiSMAwkSRgGkiQMA0kSPYZBkheSPJ3kB0kmWu3KJEeSnGjPq1s9Se5LMpnkqSTXdL3O9tb/RJLtXfUPtdefbNum3zsqSZrZXI4M/nVVfaCqxtv6LuBoVW0AjrZ1gJuADe2xE7gfOuEB3A1cB1wL3H0uQFqfT3Ztt+Wi90iSNGeXMk20FdjblvcCt3bV91XHMWBVkquBG4EjVXWmqs4CR4Atre2KqjpWVQXs63otSdIA9BoGBfxFkieS7Gy1NVX1clt+BVjTltcCL3Vte7LVLlQ/OU39bZLsTDKRZGJqaqrHoUuSZrOyx37/sqpOJfk14EiSH3U3VlUlqf4P7/9XVbuB3QDj4+Pz/vMkabno6cigqk6159PAt+nM+b/apnhoz6db91PAuq7NR1vtQvXRaeqSpAGZNQyS/EqSf3JuGdgM/A1wEDh3RdB24KG2fBDY1q4q2gS83qaTDgObk6xuJ443A4db2xtJNrWriLZ1vZYkaQB6mSZaA3y7Xe25EviTqvrzJI8DB5LsAF4Ebmv9DwE3A5PAm8AdAFV1JskXgMdbv89X1Zm2/CngG8DlwCPtIUkakFnDoKqeB94/Tf0nwA3T1Au4c4bX2gPsmaY+Abyvh/FKkuaB30CWJBkGkiTDQJKEYSBJwjCQJGEYSJIwDCRJGAaSJAwDSRKGgSQJw0CShGEgScIwkCRhGEiSMAwkSRgGkiQMA0kShoEkCcNAkoRhIEnCMJAkYRhIkjAMtAiN7XqYsV0PD3sY0pJiGEiSDANJkmEgSWIOYZBkRZInk3ynra9P8liSySTfSnJZq7+zrU+29rGu17ir1Z9LcmNXfUurTSbZ1b/dkyT1Yi5HBp8Bnu1a/zJwb1W9BzgL7Gj1HcDZVr+39SPJRuB24L3AFuBrLWBWAF8FbgI2Ah9vfSVJA9JTGCQZBW4Bvt7WA1wPPNi67AVubctb2zqt/YbWfyuwv6p+WlU/BiaBa9tjsqqer6qfAftbX0nSgPR6ZPBHwO8B/9DW3w28VlVvtfWTwNq2vBZ4CaC1v976/2P9vG1mqkuSBmTWMEjyEeB0VT0xgPHMNpadSSaSTExNTQ17OJK0ZPRyZPBh4KNJXqAzhXM98BVgVZKVrc8ocKotnwLWAbT2dwE/6a6ft81M9bepqt1VNV5V4yMjIz0MXZLUi1nDoKruqqrRqhqjcwL4u1X1O8CjwMdat+3AQ235YFuntX+3qqrVb29XG60HNgDfAx4HNrSrky5rP+NgX/ZOktSTlbN3mdHvA/uTfBF4Enig1R8A/jjJJHCGzps7VXU8yQHgGeAt4M6q+jlAkk8Dh4EVwJ6qOn4J45IkzdGcwqCq/hL4y7b8PJ0rgc7v8/fAb8+w/ZeAL01TPwQcmstYJEn94zeQJUmGgSTJMJAkYRhIkjAMJEkYBpIkDANJEoaBJAnDQJKEYSBJwjCQJGEYSJIwDCRJGAaSJAwDSRKGgSQJw0CShGEgScIwkCRhGEiSMAwkSRgGkiQMA0kShoEkCcNAkoRhIEnCMJAk0UMYJPnlJN9L8sMkx5P851Zfn+SxJJNJvpXkslZ/Z1ufbO1jXa91V6s/l+TGrvqWVptMsqv/uylJupBejgx+ClxfVe8HPgBsSbIJ+DJwb1W9BzgL7Gj9dwBnW/3e1o8kG4HbgfcCW4CvJVmRZAXwVeAmYCPw8dZXkjQgs4ZBdfyftvqO9ijgeuDBVt8L3NqWt7Z1WvsNSdLq+6vqp1X1Y2ASuLY9Jqvq+ar6GbC/9ZUkDUhP5wzaJ/gfAKeBI8D/Al6rqrdal5PA2ra8FngJoLW/Dry7u37eNjPVpxvHziQTSSampqZ6GbokqQc9hUFV/byqPgCM0vkk/xvzOqqZx7G7qsaranxkZGQYQ5CkJWlOVxNV1WvAo8BvAquSrGxNo8CptnwKWAfQ2t8F/KS7ft42M9UlSQPSy9VEI0lWteXLgX8LPEsnFD7Wum0HHmrLB9s6rf27VVWtfnu72mg9sAH4HvA4sKFdnXQZnZPMB/uxc5Kk3qycvQtXA3vbVT+/BByoqu8keQbYn+SLwJPAA63/A8AfJ5kEztB5c6eqjic5ADwDvAXcWVU/B0jyaeAwsALYU1XH+7aHkqRZzRoGVfUU8MFp6s/TOX9wfv3vgd+e4bW+BHxpmvoh4FAP45UkzQO/gSxJMgwkSYaBJAnDQJKEYSBJwjCQJGEYaBkY2/UwY7seHvYwpAXNMJD6yODRYmUYSJIMA0mSYSBJwjCQJGEYSMClnfj1pLGWAsNAkmQYSJIMA0kShoEkCcNAkoRhIEnCMJAkYRhIF+R3CLRcGAaSJMNAkmQYSJIwDCRJGAaSJHoIgyTrkjya5Jkkx5N8ptWvTHIkyYn2vLrVk+S+JJNJnkpyTddrbW/9TyTZ3lX/UJKn2zb3Jcl87KwkaXq9HBm8BXy2qjYCm4A7k2wEdgFHq2oDcLStA9wEbGiPncD90AkP4G7gOuBa4O5zAdL6fLJruy2XvmuSpF7NGgZV9XJVfb8t/x3wLLAW2Arsbd32Are25a3Avuo4BqxKcjVwI3Ckqs5U1VngCLCltV1RVceqqoB9Xa8lSRqAOZ0zSDIGfBB4DFhTVS+3pleANW15LfBS12YnW+1C9ZPT1Kf7+TuTTCSZmJqamsvQJUkX0HMYJPlV4E+B362qN7rb2if66vPY3qaqdlfVeFWNj4yMzPeP0xLnt4ulX+gpDJK8g04QfLOq/qyVX21TPLTn061+CljXtfloq12oPjpNXZI0IL1cTRTgAeDZqvrDrqaDwLkrgrYDD3XVt7WrijYBr7fppMPA5iSr24njzcDh1vZGkk3tZ23rei1J0gCs7KHPh4FPAE8n+UGrfQ64BziQZAfwInBbazsE3AxMAm8CdwBU1ZkkXwAeb/0+X1Vn2vKngG8AlwOPtIe0oJybUnrhnluGPBKp/2YNg6r6a2Cm6/5vmKZ/AXfO8Fp7gD3T1CeA9802FknS/PAbyJIkw0CSZBhIkjAMJEkYBpIkDANJEoaBJAnDQJKEYSAtSN5ET4NmGGhJ8s1UmhvDQJJkGEiSDANpwXBqS8NkGEiLlOGhfjIMJEmGgTQI3Z/i/USvhcgwkCQZBpIkw0CShGEgLWkznZ/wvIXOZxhIi4hv4povhoG0zBkwAsNAkoRhIEnCMJAkASuHPQBJC1/3OYUX7rlliCPRfJn1yCDJniSnk/xNV+3KJEeSnGjPq1s9Se5LMpnkqSTXdG2zvfU/kWR7V/1DSZ5u29yXJP3eSUnShfUyTfQNYMt5tV3A0araABxt6wA3ARvaYydwP3TCA7gbuA64Frj7XIC0Pp/s2u78nyVJmmezhkFV/RVw5rzyVmBvW94L3NpV31cdx4BVSa4GbgSOVNWZqjoLHAG2tLYrqupYVRWwr+u1JEkDcrHnDNZU1ctt+RVgTVteC7zU1e9kq12ofnKauqSLdG5+f9Bz+72cV/Dcw8J1yVcTtU/01YexzCrJziQTSSampqYG8SMlaVm42DB4tU3x0J5Pt/opYF1Xv9FWu1B9dJr6tKpqd1WNV9X4yMjIRQ5dknS+iw2Dg8C5K4K2Aw911be1q4o2Aa+36aTDwOYkq9uJ483A4db2RpJN7SqibV2vJUkakFnPGST578BvAVclOUnnqqB7gANJdgAvAre17oeAm4FJ4E3gDoCqOpPkC8Djrd/nq+rcSelP0bli6XLgkfaQtAh4T6OlY9YwqKqPz9B0wzR9C7hzhtfZA+yZpj4BvG+2cUhanOYaGJ5kHg6/gSxpWn7qX168N5EkyTCQJDlNJGkR8rxC/xkGkoZuPs5PGBhz4zSRJMkjA0nLl0cPv+CRgSTJMJAkOU0kaQmZry/KDeu24INkGEhaFPr1Rm9gTM9pImmejO162Fs6aNEwDBYw30zmxj8vLXQL+e+o00QaOC/n02Kw2Kd95sow0IJneGip6w6eYYWQYaCh8o1ei1kvb9yL5QjDMNDb9PIG3a8+g7ZY/mFKg2YYLABz+XRxoX6X8ua70C/bk5a7+f4gYxjMoF9/8PP9Cxzkfyk4089aagHg0YOWI8NgHsz0ZtLLSaK5HiX0Y5yX2kcXz+DRQrEsw2Cu/wBnehNfCFcAzGShzdfPNXgWwpgvZJC/74X2d0tL07IMg24L+Q29X/x0v/Qt1b+7GpxlHwZamAyw/jMwdCHejkLSP1rIt0vQ/PLIQNKc9OvCA0NnYTEMJC1YBs/gGAaSRO9Xsy2W7yDN1YI5Z5BkS5Lnkkwm2TXs8UjSxerXuZdBnsNZEGGQZAXwVeAmYCPw8SQbhzsqSVo+FkQYANcCk1X1fFX9DNgPbB3ymCRp2UhVDXsMJPkYsKWq/kNb/wRwXVV9+rx+O4GdbfXXgecGOtD+uAr422EPYsCW2z67v0vfYt3nf1ZVI9M1LKoTyFW1G9g97HFciiQTVTU+7HEM0nLbZ/d36VuK+7xQpolOAeu61kdbTZI0AAslDB4HNiRZn+Qy4Hbg4JDHJEnLxoKYJqqqt5J8GjgMrAD2VNXxIQ9rvizqaa6LtNz22f1d+pbcPi+IE8iSpOFaKNNEkqQhMgwkSYbBoCX5bJJKclVbT5L72m04nkpyzbDH2A9J/kuSH7V9+naSVV1td7X9fS7JjcMcZz8th1uqJFmX5NEkzyQ5nuQzrX5lkiNJTrTn1cMeaz8lWZHkySTfaevrkzzWftffahe+LGqGwQAlWQdsBv53V/kmYEN77ATuH8LQ5sMR4H1V9S+A/wncBdBuM3I78F5gC/C1djuSRW0Z3VLlLeCzVbUR2ATc2fZzF3C0qjYAR9v6UvIZ4Nmu9S8D91bVe4CzwI6hjKqPDIPBuhf4PaD7rP1WYF91HANWJbl6KKPro6r6i6p6q60eo/PdEejs7/6q+mlV/RiYpHM7ksVuWdxSpaperqrvt+W/o/MGuZbOvu5t3fYCtw5nhP2XZBS4Bfh6Ww9wPfBg67Ik9tcwGJAkW4FTVfXD85rWAi91rZ9staXk3wOPtOWlur9Ldb9mlGQM+CDwGLCmql5uTa8Aa4Y0rPnwR3Q+xP1DW3838FrXh50l8bteEN8zWCqS/A/gn07T9AfA5+hMES0ZF9rfqnqo9fkDOlML3xzk2DS/kvwq8KfA71bVG50Pyx1VVUmWxDXrST4CnK6qJ5L81rDHM58Mgz6qqn8zXT3JPwfWAz9s/2hGge8nuZZFfCuOmfb3nCT/DvgIcEP94gsti3Z/Z7FU9+ttkryDThB8s6r+rJVfTXJ1Vb3cpjlPD2+EffVh4KNJbgZ+GbgC+Aqd6dyV7ehgSfyunSYagKp6uqp+rarGqmqMzmHlNVX1Cp3bbmxrVxVtAl7vOtxetJJsoXNo/dGqerOr6SBwe5J3JllP58T594Yxxj5bFrdUafPlDwDPVtUfdjUdBLa35e3AQ4Me23yoqruqarT9u70d+G5V/Q7wKPCx1m1J7K9HBsN3CLiZzonUN4E7hjucvvmvwDuBI+1o6FhV/ceqOp7kAPAMnemjO6vq50McZ18so1uqfBj4BPB0kh+02ueAe4ADSXYALwK3DWl8g/L7wP4kXwSepBOQi5q3o5AkOU0kSTIMJEkYBpIkDANJEoaBJAnDQJKEYSBJAv4fHuqH/wtGokkAAAAASUVORK5CYII=\n","text/plain":["<Figure size 432x288 with 1 Axes>"]},"metadata":{"tags":[],"needs_background":"light"}}]},{"cell_type":"code","metadata":{"id":"1YHGdPMsJmvZ","executionInfo":{"status":"ok","timestamp":1622745878011,"user_tz":-540,"elapsed":647212,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}}},"source":["!git add ."],"execution_count":85,"outputs":[]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"hjoge75BJ3FI","executionInfo":{"status":"ok","timestamp":1622745884493,"user_tz":-540,"elapsed":2937,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"97b9b5fc-2e0d-4aed-9d41-a262a72de3bb"},"source":["!git commit -m \"all about problem 2 except the directory problem1 which is for problem 1\""],"execution_count":86,"outputs":[{"output_type":"stream","text":["[master (root-commit) 2483809] all about problem 2 except the directory problem1 which is for problem 1\n"," 27 files changed, 2617880 insertions(+)\n"," create mode 100644 binfo1-datapack1/CLIP-35L33G.bam\n"," create mode 100644 binfo1-datapack1/CLIP-35L33G.bam.bai\n"," create mode 100644 binfo1-datapack1/RNA-control.bam\n"," create mode 100644 binfo1-datapack1/RNA-control.bam.bai\n"," create mode 100644 binfo1-datapack1/RNA-siLin28a.bam\n"," create mode 100644 binfo1-datapack1/RNA-siLin28a.bam.bai\n"," create mode 100644 binfo1-datapack1/RNA-siLuc.bam\n"," create mode 100644 binfo1-datapack1/RNA-siLuc.bam.bai\n"," create mode 100644 binfo1-datapack1/RPF-siLin28a.bam\n"," create mode 100644 binfo1-datapack1/RPF-siLin28a.bam.bai\n"," create mode 100644 binfo1-datapack1/RPF-siLuc.bam\n"," create mode 100644 binfo1-datapack1/RPF-siLuc.bam.bai\n"," create mode 100644 binfo1-datapack1/filtered-RPF-siLuc.bam\n"," create mode 100644 binfo1-datapack1/fivepcounts-RPF-siLuc.bed\n"," create mode 100644 binfo1-datapack1/fivepcounts-filtered-RPF-siLuc.txt\n"," create mode 100644 binfo1-datapack1/gencode-exons-containing-startcodon.bed\n"," create mode 100644 binfo1-datapack1/gencode-plusexon.gtf\n"," create mode 100644 binfo1-datapack1/gencode-start.gtf\n"," create mode 100644 binfo1-datapack1/gencode.gtf\n"," create mode 100644 binfo1-datapack1/problem1/.DS_Store\n"," create mode 100644 binfo1-datapack1/problem1/dat/.Rhistory\n"," create mode 100644 binfo1-datapack1/problem1/dat/read-counts.txt\n"," create mode 100644 binfo1-datapack1/problem1/rst/myplot.png\n"," create mode 100644 binfo1-datapack1/problem1/src/.DS_Store\n"," create mode 100644 binfo1-datapack1/problem1/src/.Rapp.history\n"," create mode 100644 binfo1-datapack1/problem1/src/Termproj1.R\n"," create mode 100644 binfo1-datapack1/result.png\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"id":"Tf3mNvGSMs0p","executionInfo":{"status":"ok","timestamp":1622747530861,"user_tz":-540,"elapsed":458,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}}},"source":["!git remote rm origin"],"execution_count":123,"outputs":[]},{"cell_type":"code","metadata":{"id":"EvUKzuQ0KKHG","executionInfo":{"status":"ok","timestamp":1622747531746,"user_tz":-540,"elapsed":887,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}}},"source":["!git remote add origin https://github.com/JaehongYu/Bioinfo1-jhyu.git"],"execution_count":124,"outputs":[]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"ea_xfqVzK07l","executionInfo":{"status":"ok","timestamp":1622747532210,"user_tz":-540,"elapsed":466,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"29d48831-0280-401b-b1f3-29dbe9c783ba"},"source":["!git push -u origin master"],"execution_count":125,"outputs":[{"output_type":"stream","text":["fatal: could not read Username for 'https://github.com': No such device or address\n"],"name":"stdout"}]},{"cell_type":"code","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"id":"tZuR9XOKSfn1","executionInfo":{"status":"ok","timestamp":1622747588700,"user_tz":-540,"elapsed":867,"user":{"displayName":"­유재홍 / 학생 / 협동과정 생물정보학전공","photoUrl":"","userId":"03673509457096226080"}},"outputId":"54bbac60-9856-4b60-e6cf-0bbfc035ed15"},"source":["!git status"],"execution_count":126,"outputs":[{"output_type":"stream","text":["On branch master\n","Untracked files:\n","  (use \"git add <file>...\" to include in what will be committed)\n","\n","\t\u001b[31m../CoLab_TermProj.ipynb\u001b[m\n","\t\u001b[31m../CoLab_TermProj_M2.ipynb\u001b[m\n","\t\u001b[31m../colab-biolab/\u001b[m\n","\n","nothing added to commit but untracked files present (use \"git add\" to track)\n"],"name":"stdout"}]}]}